Date: 5th Dec 2024

Calculating the depth per site, such that the total depth should be the same as that in the mpileup.

For myself: Note, first I had to uncompress one of the bam files. Had previously used ```bgzip``` in the scamper module tabix to compress it. ```bgzip -d filename.bam.gz``` to decompress. 

For the mpileup (used by Grenedalf) I excluded duplicated reads, and had already excluded unmapped reads (when making the bam). However, it may have included secondary (and therefore, due to my bwa mem command, also split reads) and QCFAIL reads (unlikely I presume, given it excluded anything with BQ of <30 and MAPQ <20). Note that I am not entirely clear as the different manual versions specify different things! 

I used samtools v1.17 to create the SL_PS mpileup:
```
--ff, --excl-flags STR|INT
Filter flags: skip reads with any of the mask bits set [UNMAP,SECONDARY,QCFAIL,DUP]. Note this does not override the --incl-flags option.
```
However, a later version of the manual (v1.21) states:
```
Unmapped reads are not considered and are always discarded. By default secondary alignments, QC failures and duplicate reads will be omitted, along with low quality bases and some reads in high depth regions. See the --ff, -Q and -d options for changing this.

--ff, --excl-flags STR|INT
Filter flags: skip reads with any of the mask bits set. This defaults to SECONDARY,QCFAIL,DUP. The option is not accumulative, so specifying e.g. --ff QCFAIL will reenable output of secondary and duplicate alignments. Note this does not override the --incl-flags option.
```

In contrast, for samtools v1.17, depth automatically excludes secondary, duplicates etc:
```
-g FLAGS
By default, reads that have any of the flags UNMAP, SECONDARY, QCFAIL, or DUP set are skipped. To include these reads back in the analysis, use this option together with the desired flag or flag combination. FLAGS can be specified in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/), in octal by beginning with `0' (i.e. /^0[0-7]+/), as a decimal number not beginning with '0' or as a comma-separated list of flag names. [0]
```

Therefore, to use the total depth per site to calculate the reference allele frequency I think I need to make my depth file to also exclude duplicates, but to include secondary hits. The default is for samtools depth to exclude both. 

Note that as the bams are now all on MARS, I will be using samtools v1.18..... to be on the safe side I am going to use both -g and -G. 
```
-g FLAGS
By default, reads that have any of the flags UNMAP, SECONDARY, QCFAIL, or DUP set are skipped. To include these reads back in the analysis, use this option together with the desired flag or flag combination. FLAGS can be specified in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/), in octal by beginning with `0' (i.e. /^0[0-7]+/), as a decimal number not beginning with '0' or as a comma-separated list of flag names. [0]

For a list of flag names see samtools-flags(1).

-G FLAGS, --excl-flags FLAGS
Discard reads that have any of the flags specified by FLAGS set. FLAGS are specified as for the -g option. [UNMAP,SECONDARY,QCFAIL,DUP]
```
Note that mpileup also only counts the first read in a pair if overlapping with second read in a pair, while depth counts both. Can switch this off in depth using -s. There may still be some differences due to differences in handling paired end reads. Mpileup also counts deletions, while depth omits them. However, grenedalf is outputting the count of the reference allele, so is not bothered by deletions. 

For the bam list: 

I copied the correct order bam list from the excel file into MARS. 
```
/home/jenni/data_folder/selection_lines_poolseq_bams/16-11098_F0.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/10-13796_postCTL1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/14-13775_postCTL1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/13-13804_postCTL1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/1-13981_CTL2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/4-13844_postCTL2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/7-13989_postCTL2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/3-14949_postCTL3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/1-14031_CTL3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/2-14036_postCTL3.merged.sorted.marked.bam
No F1_I_L1
/home/jenni/data_folder/selection_lines_poolseq_bams/12-13788_postIVM1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/15-13766_postIVM1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/3-13982_postIVM2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/6-13911_postIVM2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/9-14008_postIVM2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/6-14917_postIVM3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/5-14916_postIVM3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/4-14907_postIVM3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/11-13740_postMOX1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/13-13807_postMOX1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/14-13723_postMOX1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/2-13934_postMOX2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/5-13918_postMOX2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/8-13930_postMOX2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/9-14954_postMOX3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/7-14029_postMOX3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/8-14953_postMOX3.merged.sorted.marked.bam
```
I then removed the No F1_I_L1 line and used sed to update for MARS: 
```
sed 's/\/home\/jenni\/data_folder\/selection_lines_poolseq_bams/\/users\/jmi45g\/project0005\/HCON\/DATA\/SLPS_Illumina\/Aligned_bam_files\/bam_files/g' SL_only_Dec24_bam.list > tmp
```

```
#!/bin/bash -l

############# SLURM SETTINGS #############
#SBATCH --account=project0005   # account name (mandatory), if the job runs under a project then it'll be the project name, if not then it should =none
#SBATCH --job-name=depth_Dec24     # some descriptive job name of your choice
#SBATCH --output=%x-%J.out      # output file name will contain job name + job ID
#SBATCH --error=%x-%J.err       # error file name will contain job name + job ID
#SBATCH --time=0-05:00:00       # time limit for the whole run, in the form of d-hh:mm:ss, also accepts mm, mm:ss, hh:mm:ss, d-hh, d-hh:mm
#SBATCH --mem=5G                # memory required per node, in the form of [num][M|G|T]
#SBATCH --ntasks=1              # number of Slurm tasks to be launched, increase for multi-process runs ex. MPI
#SBATCH --cpus-per-task=1       # number of processor cores to be assigned for each task, default is 1, increase for multi-threaded runs
#SBATCH --ntasks-per-node=1     # number of tasks to be launched on each allocated node

############# LOADING MODULES (optional) #############
#module load apps/xxx
#module load libs/xxx

module load apps/samtools/1.18/gcc-4.8.5

############# MY CODE #############

samtools depth -a -f /users/jmi45g/project0005/HCON/DATA/SLPS_Illumina/Aligned_bam_files/bam_files/SL_only_Dec24_bam.list -H  -o SL_only_Dec_24_Chr5_30-45Mb.depth -q 30 -Q 20 -r hcontortus_chr5_Celeg_TT_arrow_pilon:30000000-45000000 -g SECONDARY,QCFAIL -G DUP -s

# -a Output all positions (including those with zero depth)
# -f FILE Use the BAM files specified in the FILE (a file of filenames, one file per line) []
# -H Write a comment line showing column names at the beginning of the output. The names are CHROM, POS, and then the input file name for each depth column. If one of the inputs came from stdin, the name “-” will be used for the corresponding column.
# -o FILE Write output to FILE. Using “-” for FILE will send the output to stdout (also the default if this option is not used).
# -q, --min-BQ INT Only count reads with base quality greater than or equal to INT
# -Q, --min-MQ INT Only count reads with mapping quality greater than or equal to INT
# -r CHR:FROM-TO Only report depth in specified region.
# -g FLAGS By default, reads that have any of the flags UNMAP, SECONDARY, QCFAIL, or DUP set are skipped. To include these reads back in the analysis, use this option together with the desired flag or flag combination. FLAGS can be specified in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/), in octal by beginning with `0' (i.e. /^0[0-7]+/), as a decimal number not beginning with '0' or as a comma-separated list of flag names. [0] For a list of flag names see samtools-flags(1).
# -G FLAGS, --excl-flags FLAGS Discard reads that have any of the flags specified by FLAGS set. FLAGS are specified as for the -g option. [UNMAP,SECONDARY,QCFAIL,DUP]
# -s For the overlapping section of a read pair, count only the bases of the first read. Note this algorithm changed in 1.13 so the results may differ slightly to older releases.
```
