Using the 57X subsampled mpileup file.
```
# Calculate Fst in 10kb windows along the genome:

module load grenedalf/0.3.0


grenedalf fst \
--sync-path /home/jenni/data_folder/working_folder/selection_lines_poolseq/mpileup/Jan24_correct_order_syncs/SLps_only_co_Hcwbps18_q20Q30_noindels_ss57.java.sync \
--reference-genome-fasta-file /home/jenni/data_folder/wbps18/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa \
--rename-samples-file /home/jenni/data_folder/working_folder/selection_lines_poolseq/pairwise_comparisons/LINES_IN_CORRECT_ORDER/grenedalf/Jan24_F0_vs_all_sel_lines_only/SL_ss57_sample_names.txt \
--comparand-list /home/jenni/data_folder/working_folder/selection_lines_poolseq/pairwise_comparisons/LINES_IN_CORRECT_ORDER/grenedalf/Jan24_F0_vs_all_sel_lines_only/SL_pairwise_comparisons.txt \
--filter-sample-min-count 2 \
--filter-sample-min-coverage 57 \
--filter-sample-max-coverage 57 \
--window-type sliding \
--window-sliding-width 10000 \
--window-sliding-stride 10000 \
--method unbiased-nei \
--pool-sizes 400 \
--omit-na-windows \
--separator-char tab \
--out-dir /home/jenni/data_folder/working_folder/selection_lines_poolseq/pairwise_comparisons/LINES_IN_CORRECT_ORDER/grenedalf/Jan24_F0_vs_all_sel_lines_only/ss57 \
--file-prefix 400_Gf_F0_vs_SL_only_Sel_lines_Hcwbps18_q20Q30_noindels_ss57_10kb_fst \
--log-file grenedalf_10kb_F0_vs_SL_only_fst_ss57.log \
2>error_log_10kb_fst_f057 &
```
Completed fine: 
```
Run the following command to get the references that need to be cited:
`grenedalf citation Czech2023-grenedalf Kofler2011-PoPoolation2`

Started 2024-01-17 13:29:08

Reading reference genome fasta
Reference genome contains 7 chromosomes
Renamed 27 out of 27 sample names.
Processing 27 samples

Computing FST between 35 pairs of samples.
At chromosome hcontortus_chr4_Celeg_TT_arrow_pilon
At chromosome hcontortus_chr5_Celeg_TT_arrow_pilon
At chromosome hcontortus_chr2_Celeg_TT_arrow_pilon
At chromosome hcontortus_chrX_Celeg_TT_arrow_pilon
At chromosome hcontortus_chr1_Celeg_TT_arrow_pilon
At chromosome hcontortus_chr3_Celeg_TT_arrow_pilon

Processed 6 chromosomes with 9076440 (non-filtered) positions in 28343 windows.
Thereof, skipped 750 windows without any FST values.
Sample filter summary (summed up across all samples):
Passed:               2373425383
Below min coverage:   55298624
Total filter summary (after applying all sample filters):
Passed:                  9076440
Not SNP:                 80876301

Finished 2024-01-17 13:31:28

```
Rename the chromosomes, so that the names are just 5 or X etc:
```
cat 400_Gf_F0_vs_SL_only_Sel_lines_Hcwbps18_q20Q30_noindels_ss57_10kb_fstfst.csv | sed 's/hcontortus_chr//g' | sed 's/_Celeg_TT_arrow_pilon//g' > tmp
mv tmp 400_Gf_F0_vs_SL_only_Sel_lines_Hcwbps18_q20Q30_noindels_ss57_10kb_fstfst.csv
```
Then plot in R: 
```
module load R/4.3.1

R
library(ggplot2)
library(reshape2)


# Note, did each sequentially, but just keeping one set of code:
win<-read.table("400_Gf_F0_vs_SL_only_Sel_lines_Hcwbps18_q20Q30_noindels_ss57_10kb_fstfst.csv", header=T, sep="\t", na.strings = "NA")
head(win)
tail(win)
summary(win)
```
```
    chrom               start               end                snps
 Length:27593       Min.   :       1   Min.   :   10000   Min.   :   1.0
 Class :character   1st Qu.:11730001   1st Qu.:11740000   1st Qu.: 192.0
 Mode  :character   Median :23690001   Median :23700000   Median : 323.0
                    Mean   :23703247   Mean   :23713246   Mean   : 328.9
                    3rd Qu.:35530001   3rd Qu.:35540000   3rd Qu.: 454.0
                    Max.   :51820001   Max.   :51830000   Max.   :1009.0

   F0.F1_C_L1         F0.F1_C_L2         F0.F1_C_L3         F0.F1_I_L2
 Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010
 1st Qu.:-0.00155   1st Qu.:-0.00164   1st Qu.:-0.00148   1st Qu.: 0.00096
 Median :-0.00038   Median :-0.00051   Median :-0.00026   Median : 0.00424
 Mean   : 0.00004   Mean   :-0.00019   Mean   : 0.00024   Mean   : 0.01016
 3rd Qu.: 0.00112   3rd Qu.: 0.00087   3rd Qu.: 0.00131   3rd Qu.: 0.01065
 Max.   : 0.06936   Max.   : 0.05660   Max.   : 0.05660   Max.   : 0.26429
 NA's   :66         NA's   :66         NA's   :60         NA's   :52
   F0.F1_I_L3         F0.F1_M_L1         F0.F1_M_L2         F0.F1_M_L3
 Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010
 1st Qu.: 0.00448   1st Qu.: 0.00305   1st Qu.:-0.00004   1st Qu.: 0.00111
 Median : 0.01150   Median : 0.00723   Median : 0.00262   Median : 0.00437
 Mean   : 0.01964   Mean   : 0.01132   Mean   : 0.00712   Mean   : 0.01024
 3rd Qu.: 0.02496   3rd Qu.: 0.01443   3rd Qu.: 0.00769   3rd Qu.: 0.01010
 Max.   : 0.33222   Max.   : 0.28621   Max.   : 0.30119   Max.   : 0.33582
 NA's   :64         NA's   :48         NA's   :69         NA's   :58
   F0.F2_C_L1         F0.F2_C_L2         F0.F2_C_L3         F0.F2_I_L1
 Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010
 1st Qu.:-0.00146   1st Qu.:-0.00147   1st Qu.:-0.00142   1st Qu.: 0.00690
 Median :-0.00025   Median :-0.00030   Median :-0.00017   Median : 0.01642
 Mean   : 0.00021   Mean   : 0.00006   Mean   : 0.00025   Mean   : 0.02785
 3rd Qu.: 0.00128   3rd Qu.: 0.00109   3rd Qu.: 0.00140   3rd Qu.: 0.03857
 Max.   : 0.08473   Max.   : 0.06051   Max.   : 0.06189   Max.   : 0.38165
 NA's   :58         NA's   :59         NA's   :60         NA's   :66
   F0.F2_I_L2         F0.F2_I_L3         F0.F2_M_L1         F0.F2_M_L2
 Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010
 1st Qu.: 0.00091   1st Qu.: 0.00132   1st Qu.: 0.00505   1st Qu.: 0.00080
 Median : 0.00458   Median : 0.00515   Median : 0.01087   Median : 0.00442
 Mean   : 0.01586   Mean   : 0.01076   Mean   : 0.01860   Mean   : 0.01539
 3rd Qu.: 0.01212   3rd Qu.: 0.01280   3rd Qu.: 0.02056   3rd Qu.: 0.01162
 Max.   : 0.49488   Max.   : 0.30119   Max.   : 0.57663   Max.   : 0.41958
 NA's   :65         NA's   :61         NA's   :42         NA's   :52
   F0.F2_M_L3         F0.F3_C_L1         F0.F3_C_L2         F0.F3_C_L3
 Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010
 1st Qu.: 0.00075   1st Qu.:-0.00148   1st Qu.:-0.00162   1st Qu.:-0.00150
 Median : 0.00451   Median :-0.00027   Median :-0.00047   Median :-0.00029
 Mean   : 0.01994   Mean   : 0.00015   Mean   :-0.00010   Mean   : 0.00014
 3rd Qu.: 0.01199   3rd Qu.: 0.00126   3rd Qu.: 0.00098   3rd Qu.: 0.00118
 Max.   : 0.93087   Max.   : 0.06275   Max.   : 0.06590   Max.   : 0.05829
 NA's   :62         NA's   :61         NA's   :64         NA's   :58
   F0.F3_I_L1         F0.F3_I_L2         F0.F3_I_L3         F0.F3_M_L1
 Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010
 1st Qu.: 0.00139   1st Qu.: 0.00703   1st Qu.: 0.00081   1st Qu.: 0.00678
 Median : 0.00530   Median : 0.01671   Median : 0.00442   Median : 0.01436
 Mean   : 0.01274   Mean   : 0.03759   Mean   : 0.01709   Mean   : 0.02470
 3rd Qu.: 0.01244   3rd Qu.: 0.03836   3rd Qu.: 0.01193   3rd Qu.: 0.02648
 Max.   : 0.55469   Max.   : 0.64633   Max.   : 0.53584   Max.   : 0.62242
 NA's   :70         NA's   :68         NA's   :63         NA's   :35
   F0.F3_M_L2         F0.F3_M_L3       F3_M_L1.F3_M_L2    F3_M_L1.F3_M_L3
 Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010
 1st Qu.: 0.00151   1st Qu.: 0.00073   1st Qu.: 0.00864   1st Qu.: 0.00744
 Median : 0.00534   Median : 0.00428   Median : 0.01702   Median : 0.01569
 Mean   : 0.02493   Mean   : 0.02285   Mean   : 0.02405   Mean   : 0.02275
 3rd Qu.: 0.01355   3rd Qu.: 0.01086   3rd Qu.: 0.03119   3rd Qu.: 0.02810
 Max.   : 0.93087   Max.   : 0.93087   Max.   : 0.28337   Max.   : 0.27306
 NA's   :65         NA's   :62         NA's   :39         NA's   :35
 F3_M_L2.F3_M_L3    F3_I_L1.F3_I_L2    F3_I_L1.F3_I_L3    F3_I_L2.F3_I_L3
 Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010
 1st Qu.: 0.00119   1st Qu.: 0.00679   1st Qu.: 0.00181   1st Qu.: 0.00520
 Median : 0.00477   Median : 0.01765   Median : 0.00693   Median : 0.01371
 Mean   : 0.00801   Mean   : 0.02801   Mean   : 0.01176   Mean   : 0.02252
 3rd Qu.: 0.01091   3rd Qu.: 0.03779   3rd Qu.: 0.01702   3rd Qu.: 0.03122
 Max.   : 0.10540   Max.   : 0.41664   Max.   : 0.21594   Max.   : 0.37350
 NA's   :60         NA's   :78         NA's   :69         NA's   :70
 F3_I_L1.F3_M_L1    F3_I_L2.F3_M_L2    F3_I_L3.F3_M_L3
 Min.   :-0.01010   Min.   :-0.01010   Min.   :-0.01010
 1st Qu.: 0.00877   1st Qu.: 0.00315   1st Qu.: 0.00020
 Median : 0.01767   Median : 0.00961   Median : 0.00305
 Mean   : 0.02321   Mean   : 0.01626   Mean   : 0.00601
 3rd Qu.: 0.03131   3rd Qu.: 0.02167   3rd Qu.: 0.00816
 Max.   : 0.25450   Max.   : 0.31473   Max.   : 0.16266
 NA's   :40         NA's   :71         NA's   :58
```
```
# Let's make Chromosome a factor not a character. 
win$chrom<-as.factor(win$chrom)
summary(win)
tail(win)
# Perfect!

# And calculate the mid-point of the window for plotting:
win$mid<-(win$end - 5000)
head(win)

win_new<-win[,c(1,4:40)]

# Now to melt the table. Need package reshape2

library(reshape2)
win_melt <- melt(win_new, id.vars=c("chrom","mid","snps"))
head(win_melt)

# To sort the table so that the chromosomes, and coordinates are in ascending order:
tmp<-win_melt
tmp2<-tmp[order(tmp$chrom, tmp$mid),]
head(tmp2)
tail(tmp2)
# MUCH better!

summary(win_melt)
```
```
                                  chrom             mid
 hcontortus_chr1_Celeg_TT_arrow_pilon:155715   Min.   :    5000
 hcontortus_chr2_Celeg_TT_arrow_pilon:162645   1st Qu.:11735000
 hcontortus_chr3_Celeg_TT_arrow_pilon:149100   Median :23695000
 hcontortus_chr4_Celeg_TT_arrow_pilon:178605   Mean   :23708246
 hcontortus_chr5_Celeg_TT_arrow_pilon:163905   3rd Qu.:35535000
 hcontortus_chrX_Celeg_TT_arrow_pilon:155785   Max.   :51825000

      snps              variable          value
 Min.   :   1.0   F0.F1_C_L1: 27593   Min.   :-0.0101
 1st Qu.: 192.0   F0.F1_C_L2: 27593   1st Qu.: 0.0002
 Median : 323.0   F0.F1_C_L3: 27593   Median : 0.0042
 Mean   : 328.9   F0.F1_I_L2: 27593   Mean   : 0.0134
 3rd Qu.: 454.0   F0.F1_I_L3: 27593   3rd Qu.: 0.0141
 Max.   :1009.0   F0.F1_M_L1: 27593   Max.   : 0.9309
                  (Other)   :800197   NA's   :2074
```
Try removing windows with 'Na' in at least one pwc. The idea being (from previous work on this by me) that this would remove some of hte lowest SNP windows. The data was indicating that having more SNPs reduced Fst, and was significant, but there was still a fair amount of variation, and hard to define an appropriate cut-off. However, if have only a few SNPs then likely to have some pwc with no SNPs and hence Na (from grenedalf wiki website: '--na-entry
TEXT=nan
Set the text to use in the output for n/a and NaN entries (e.g., resulting from positions with no counts, or windows with no variants). This is useful to match formatting expectations of downstream software.
Output'.)
```
# Filter out any windows which have 'NaN' in at least one pwc:

win2<-na.omit(win)
summary(win2)
```
```
win_new<-win2[,c(1,4:40)]

# Now to melt the table. Need package reshape2

library(reshape2)
win_melt <- melt(win_new, id.vars=c("chrom","mid","snps"))
head(win_melt)

# To sort the table so that the chromosomes, and coordinates are in ascending order:
tmp<-win_melt
tmp2<-tmp[order(tmp$chrom, tmp$mid),]
head(tmp2)
tail(tmp2)
# MUCH better!

win_melt<-tmp2
summary(win_melt)
```
NOTE that this has removed the highest Fst window - 0.9.... 
```
                                  chrom             mid
 hcontortus_chr1_Celeg_TT_arrow_pilon:154945   Min.   :    5000
 hcontortus_chr2_Celeg_TT_arrow_pilon:162155   1st Qu.:11735000
 hcontortus_chr3_Celeg_TT_arrow_pilon:148645   Median :23685000
 hcontortus_chr4_Celeg_TT_arrow_pilon:178045   Mean   :23704053
 hcontortus_chr5_Celeg_TT_arrow_pilon:162365   3rd Qu.:35527500
 hcontortus_chrX_Celeg_TT_arrow_pilon:153965   Max.   :51825000

      snps              variable          value
 Min.   :   1.0   F0.F1_C_L1: 27432   Min.   :-0.0101010
 1st Qu.: 195.0   F0.F1_C_L2: 27432   1st Qu.: 0.0001827
 Median : 325.0   F0.F1_C_L3: 27432   Median : 0.0041769
 Mean   : 330.8   F0.F1_I_L2: 27432   Mean   : 0.0134326
 3rd Qu.: 455.0   F0.F1_I_L3: 27432   3rd Qu.: 0.0141165
 Max.   :1009.0   F0.F1_M_L1: 27432   Max.   : 0.7322230
                  (Other)   :795528
```

```
# Filter the data to retain each generation separately:

library(stringr)
library(dplyr)

win_melt_F2<- win_melt %>%
  filter(str_detect(variable, "F2"))

win_melt_F3<- win_melt %>%
  filter(str_detect(variable, "F3"))

# Need to further filter to retain just the F3 v F0 pwc, as also have F3 in the IVM vs MOX 3rd gen comparisons
win_melt_F3.F0<- win_melt_F3 %>%
  filter(str_detect(variable, "F0"))

# Select the 3rd gen IVM vs MOX pwc
win_melt_selpwc<- win_melt %>%
  filter(!str_detect(variable, "F0"))

# Only want the following for those with IVM for F1 (as no F1 line 1 ivm sample)
# tmpF1<-win_melt_F1
# tmpF1$variable.new <- factor(tmpF1$variable, levels=(c("",levels(win_melt_F1$variable))))
# Note that the above doesn't work - I think I must have subset the data slightly differently before. It's basically inserting an empty level at the start. If I plot it, I get the empty variable if I have drop="FALSE". But I also get all the other filtered out variables. Hmmm.... If I do drop="TRUE" then I get only 8 plots. 
# No - it will not work with the str_detect - it 'remembers' all those removed.
# So, instead do it this way:

win_F1<-win2[,c(1,4:12,40)]
summary(win_F1)
# Now to melt the table. Need package reshape2

library(reshape2)
win_melt_F1 <- melt(win_F1, id.vars=c("chrom","mid","snps"))
head(win_melt_F1)

# To sort the table so that the chromosomes, and coordinates are in ascending order:
tmp<-win_melt_F1
tmp2<-tmp[order(tmp$chrom, tmp$mid),]
head(tmp2)
tail(tmp2)
# MUCH better!

win_melt_F1<-tmp2
summary(win_melt_F1)

# Then, add in the extra facet:
win_melt_F1$variable.new <- factor(win_melt_F1$variable, levels=(c("F0.F1_C_L1", "F0.F1_C_L2", "F0.F1_C_L3", "", "F0.F1_I_L2", "F0.F1_I_L3", "F0.F1_M_L1", "F0.F1_M_L2", "F0.F1_M_L3")))   


# And to plot the chromosomes, by colour:

pdf("400_Gf_F0_vs_F2_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf",width = 16,height = 12,useDingbats = FALSE)

# The x-axis has all data plotted from the start to the end of the file - not an exact position (as this data has the contigs too jumbled) - just a positional index
# The y-axis has the Fst value

# DON'T FORGET TO ALTER TITLES AND SUBTITLES! And change variable to variable.new when using IVM samples 

x<- ggplot(data=tmpF1, mapping=aes(x=1:nrow(tmpF1), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3) +
labs(y="10kb window Fst value", x="Positional index")+
theme(axis.title.x = element_text(size=14), axis.text.x = element_blank(), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable.new, ncol=3, drop=FALSE)
x
dev.off()



# Trying with patchwork (https://patchwork.data-imaginist.com/index.html)

install.packages('patchwork')
library(patchwork)


x<- ggplot(data=win_melt_F1, mapping=aes(x=1:nrow(win_melt_F1), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3) +
labs(y="10kb window Fst value", x="Positional index")+
theme(axis.title.x = element_text(size=14), axis.text.x = element_blank(), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable.new, ncol=3, drop=FALSE)+
ggtitle('F1 generation')


x2<- ggplot(data=win_melt_F2, mapping=aes(x=1:nrow(win_melt_F2), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3) +
labs(y="10kb window Fst value", x="Positional index")+
theme(axis.title.x = element_text(size=14), axis.text.x = element_blank(), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE)+
ggtitle('F2 generation')


x3<- ggplot(data=win_melt_F3.F0, mapping=aes(x=1:nrow(win_melt_F3.F0), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3) +
labs(y="10kb window Fst value", x="Positional index")+
theme(axis.title.x = element_text(size=14), axis.text.x = element_blank(), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE) +
ggtitle('F3 generation')

x4<- ggplot(data=win_melt_selpwc, mapping=aes(x=1:nrow(win_melt_selpwc), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3) +
labs(y="10kb window Fst value", x="Positional index")+
theme(axis.title.x = element_text(size=14), axis.text.x = element_blank(), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE) +
ggtitle('F3 generation: Selected vs Selected')

pdf("400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf",width = 16,height = 12,useDingbats = FALSE)
patchwork <- x + x2 + x3 + x4 + plot_layout(guides = 'collect')

patchwork + plot_annotation(
  title = 'Haemonchus contortus Selection Lines experiment',
   caption = 'Subsampled depth = 57',
tag_levels = 'A'
)
dev.off()
```
And to plot just ChrV, continued from above in R:
```
x<- ggplot(data=subset(win_melt_F1, win_melt_F1$chrom == "hcontortus_chr5_Celeg_TT_arrow_pilon"),  mapping=aes(x=mid/1000000, y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(colour="#619CFF", alpha=0.3) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
theme(axis.title.x = element_text(size=14), axis.text.x = element_text(size=12), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable.new, ncol=3, drop=FALSE)+
ggtitle('F1 generation')


x2<- ggplot(data=subset(win_melt_F2, win_melt_F2$chrom == "hcontortus_chr5_Celeg_TT_arrow_pilon"), mapping=aes(x=mid/1000000, y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(colour="#619CFF", alpha=0.3) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
theme(axis.title.x = element_text(size=14), axis.text.x = element_text(size=12), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE)+
ggtitle('F2 generation')


x3<- ggplot(data=subset(win_melt_F3.F0, win_melt_F3.F0$chrom == "hcontortus_chr5_Celeg_TT_arrow_pilon"), mapping=aes(x=mid/1000000, y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(colour="#619CFF", alpha=0.3) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
theme(axis.title.x = element_text(size=14), axis.text.x = element_text(size=12), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE) +
ggtitle('F3 generation')

x4<- ggplot(data=subset(win_melt_selpwc, win_melt_selpwc$chrom == "hcontortus_chr5_Celeg_TT_arrow_pilon"), mapping=aes(x=mid/1000000, y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(colour="#619CFF", alpha=0.3) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
theme(axis.title.x = element_text(size=14), axis.text.x = element_text(size=12), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE) +
ggtitle('F3 generation: Selected vs Selected')

pdf("ChrV_400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf",width = 16,height = 12,useDingbats = FALSE)
patchwork <- x + x2 + x3 + x4

patchwork + plot_annotation(
  title = 'Haemonchus contortus Selection Lines experiment : Chromosome V',
   caption = 'Subsampled depth = 57',
tag_levels = 'A'
)
dev.off()
```
```
C:
cd "C:/Users/jmi45g/OneDrive - University of Glasgow/BIOINFORMATICS/Roz's Hcon sequencing/Selection lines poolseq/F0_vs_all_Jan24/ss57 plots"

scp jenni@scamper.mvls.gla.ac.uk:/home/jenni/data_folder/working_folder/selection_lines_poolseq/pairwise_comparisons/LINES_IN_CORRECT_ORDER/grenedalf/Jan24_F0_vs_all_sel_lines_only/ss57/400_Gf_F0_vs_F2_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf ./400_Gf_F0_vs_F2_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf

scp jenni@scamper.mvls.gla.ac.uk:/home/jenni/data_folder/working_folder/selection_lines_poolseq/pairwise_comparisons/LINES_IN_CORRECT_ORDER/grenedalf/Jan24_F0_vs_all_sel_lines_only/ss57/400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf ./400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf

scp jenni@scamper.mvls.gla.ac.uk:/home/jenni/data_folder/working_folder/selection_lines_poolseq/pairwise_comparisons/LINES_IN_CORRECT_ORDER/grenedalf/Jan24_F0_vs_all_sel_lines_only/ss57/ChrV_400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf ./ChrV_400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf
```


# Calculate the u+5sd for the cut-off to find the window.

Using the R code above to load, manipulate and plot the data. 

1. Try using the F0vCTL (all generations) u+5sd for the cut-off and see if looks helpful. - No, everything is high.
2. Try using the F0vsF3_M for M and F0vsF3_I as are different anthelmintics/selection. ie. Wouldn't want to use the same cut-off for both Choi and a farm, or for btub and ivm. 

```
module load R/4.3.1

R
library(ggplot2)
library(reshape2)

# Complete the above of manipulating the table etc.

win_melt_CTL<- win_melt %>%
  filter(str_detect(variable, "C"))

u<-mean(win_melt_CTL$value)
stdev<-sd(win_melt_CTL$value)

u5sd<-(u + (5*stdev))

u
# 8.132433e-05
stdev
# 0.002867573
u5sd
# 0.01441919
```

2. The u+5sd for MOX F3:
```
win_melt_F3.M<- win_melt_F3 %>%
  filter(str_detect(variable, "M"))

  u<-mean(win_melt_F3.M$value)
stdev<-sd(win_melt_F3.M$value)

u5sd<-(u + (5*stdev))

u
# 0.01919577
stdev
# 0.03904694
u5sd
# 0.2144305
```

3. The u+5sd for IVM F3:
```
win_melt_F3.I<- win_melt_F3 %>%
  filter(str_detect(variable, "I"))

  u<-mean(win_melt_F3.I$value)
stdev<-sd(win_melt_F3.I$value)

u5sd<-(u + (5*stdev))

u
# 0.01948157
stdev
# 0.03292054
u5sd
# 0.1840843
```  
```
# Plot the plots with lines to show the cut-off. And plot the top part twice so that it is darker maybe for the ChrV plots.

# Trying with patchwork (https://patchwork.data-imaginist.com/index.html)

library(patchwork)


x<- ggplot(data=win_melt_F1, mapping=aes(x=1:nrow(win_melt_F1), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=1) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_I_L1","F0.F1_I_L2", "F0.F1_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_M_L1","F0.F1_M_L2", "F0.F1_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=14), axis.text.x = element_blank(), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable.new, ncol=3, drop=FALSE)+
ggtitle('F1 generation')
# geom_hline(data = data.frame(yint = 0.1840843, variable.new=c("F0.F1_I_L1","F0.F1_I_L2", "F0.F1_I_L3")), aes(yintercept = yint), linetype = "dashed", linewidth = 0.5, colour = "black") +
# geom_hline(data = data.frame(yint = 0.2144305, variable.new=c("F0.F1_M_L1","F0.F1_M_L2", "F0.F1_M_L3")), aes(yintercept = yint), linetype = "dotted", linewidth = 0.5, colour = "black") +



x2<- ggplot(data=win_melt_F2, mapping=aes(x=1:nrow(win_melt_F2), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_I_L1","F0.F2_I_L2", "F0.F2_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_M_L1","F0.F2_M_L2", "F0.F2_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=14), axis.text.x = element_blank(), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE)+
ggtitle('F2 generation')


x3<- ggplot(data=win_melt_F3.F0, mapping=aes(x=1:nrow(win_melt_F3.F0), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_I_L1","F0.F3_I_L2", "F0.F3_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_M_L1","F0.F3_M_L2", "F0.F3_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=14), axis.text.x = element_blank(), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE) +
ggtitle('F3 generation')


x4<- ggplot(data=win_melt_selpwc, mapping=aes(x=1:nrow(win_melt_selpwc), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(yintercept = 0.1840843, linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(yintercept = 0.2144305, linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=14), axis.text.x = element_blank(), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE) +
ggtitle('F3 generation: Selected vs Selected')

pdf("u5sd_400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf",width = 16,height = 12,useDingbats = FALSE)
patchwork <- x + x2 + x3 + x4 + plot_layout(guides = 'collect')

patchwork + plot_annotation(
  title = 'Haemonchus contortus Selection Lines experiment',
   caption = 'Subsampled depth = 57, Only sites with variants in all samples included, Dashed hline = F0.F3_I u+5sd,  Dotted hline = F0.F3_M u+5sd',
tag_levels = 'A'
)
dev.off()
```

And to plot just ChrV, continued from above in R:
```
x<- ggplot(data=subset(win_melt_F1, win_melt_F1$chrom == "hcontortus_chr5_Celeg_TT_arrow_pilon"),  mapping=aes(x=mid/1000000, y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(colour="#619CFF", alpha=0.3) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_I_L1","F0.F1_I_L2", "F0.F1_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_M_L1","F0.F1_M_L2", "F0.F1_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=14), axis.text.x = element_text(size=12), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable.new, ncol=3, drop=FALSE)+
ggtitle('F1 generation')


x2<- ggplot(data=subset(win_melt_F2, win_melt_F2$chrom == "hcontortus_chr5_Celeg_TT_arrow_pilon"), mapping=aes(x=mid/1000000, y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(colour="#619CFF", alpha=0.3) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_I_L1","F0.F2_I_L2", "F0.F2_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_M_L1","F0.F2_M_L2", "F0.F2_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=14), axis.text.x = element_text(size=12), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE)+
ggtitle('F2 generation')


x3<- ggplot(data=subset(win_melt_F3.F0, win_melt_F3.F0$chrom == "hcontortus_chr5_Celeg_TT_arrow_pilon"), mapping=aes(x=mid/1000000, y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(colour="#619CFF", alpha=0.3) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_I_L1","F0.F3_I_L2", "F0.F3_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_M_L1","F0.F3_M_L2", "F0.F3_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=14), axis.text.x = element_text(size=12), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE) +
ggtitle('F3 generation')

x4<- ggplot(data=subset(win_melt_selpwc, win_melt_selpwc$chrom == "hcontortus_chr5_Celeg_TT_arrow_pilon"), mapping=aes(x=mid/1000000, y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(colour="#619CFF", alpha=0.3) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
geom_hline(yintercept = 0.1840843, linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(yintercept = 0.2144305, linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=14), axis.text.x = element_text(size=12), axis.title.y = element_text(size=14), axis.text.y = element_text(size=12), strip.text.x = element_text(size = 14))+
facet_wrap(~variable, ncol=3, drop=TRUE) +
ggtitle('F3 generation: Selected vs Selected')

pdf("u5sd_ChrV_400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf",width = 16,height = 12,useDingbats = FALSE)
patchwork <- x + x2 + x3 + x4

patchwork + plot_annotation(
  title = 'Haemonchus contortus Selection Lines experiment : Chromosome V',
   caption = 'Subsampled depth = 57, Only sites with variants in all samples included, Dashed hline = F0.F3_I u+5sd,  Dotted hline = F0.F3_M u+5sd',
tag_levels = 'A'
)
dev.off()
```
```
C:
cd "C:/Users/jmi45g/OneDrive - University of Glasgow/BIOINFORMATICS/Roz's Hcon sequencing/Selection lines poolseq/F0_vs_all_Jan24/ss57 plots"

scp jenni@scamper.mvls.gla.ac.uk:/home/jenni/data_folder/working_folder/selection_lines_poolseq/pairwise_comparisons/LINES_IN_CORRECT_ORDER/grenedalf/Jan24_F0_vs_all_sel_lines_only/ss57/u5sd_400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf ./u5sd_400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf

scp jenni@scamper.mvls.gla.ac.uk:/home/jenni/data_folder/working_folder/selection_lines_poolseq/pairwise_comparisons/LINES_IN_CORRECT_ORDER/grenedalf/Jan24_F0_vs_all_sel_lines_only/ss57/u5sd_ChrV_400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf ./u5sd_ChrV_400_Gf_SLps_Hcwbps18_q20Q30_ss57_sellinesonly.10kb_fst_jan2024.pdf
```

And to get the actual window start and stop points: 

1. For IVM:
```
# In R, continuing from above:

tmp<-subset(win_melt_F3.F0, win_melt_F3.F0$value > 0.1840843)
win_ivm<-tmp %>%
  filter(str_detect(variable, "I"))
cor(win_ivm$value, win_ivm$snps)
#[1] 0.04506447


tmp<-subset(win_melt_F3.F0, win_melt_F3.F0$value > 0.2144305)
win_mox<-tmp %>%
  filter(str_detect(variable, "M"))
cor(win_mox$value, win_mox$snps)
#[1] -0.05622397

write.table(win_ivm, file="F3_IVM_Fst_u5sd_loci.txt", row.names=FALSE, sep="\t", quote= FALSE)
write.table(win_mox, file="F3_MOX_Fst_u5sd_loci.txt", row.names=FALSE, sep="\t", quote= FALSE)
```

```
> summary(tmp)
                                  chrom           mid                snps      
 hcontortus_chr1_Celeg_TT_arrow_pilon:   4   Min.   :  125000   Min.   :  4.0  
 hcontortus_chr2_Celeg_TT_arrow_pilon:   0   1st Qu.:36625000   1st Qu.:173.0  
 hcontortus_chr3_Celeg_TT_arrow_pilon:   0   Median :39605000   Median :279.0  
 hcontortus_chr4_Celeg_TT_arrow_pilon:   0   Mean   :39264661   Mean   :288.1  
 hcontortus_chr5_Celeg_TT_arrow_pilon:4562   3rd Qu.:42627500   3rd Qu.:390.0  
 hcontortus_chrX_Celeg_TT_arrow_pilon:  34   Max.   :48795000   Max.   :752.0  
                                                                               
       variable        value       
 F0.F3_I_L2:1252   Min.   :0.1841  
 F0.F3_M_L2:1196   1st Qu.:0.2162  
 F0.F3_M_L3:1026   Median :0.2595  
 F0.F3_I_L3: 649   Mean   :0.2862  
 F0.F3_M_L1: 332   3rd Qu.:0.3402  
 F0.F3_I_L1: 145   Max.   :0.7322  
 (Other)   :   0                   


> summary(win_ivm)
                                  chrom           mid                snps      
 hcontortus_chr1_Celeg_TT_arrow_pilon:   4   Min.   :  125000   Min.   :  5.0  
 hcontortus_chr2_Celeg_TT_arrow_pilon:   0   1st Qu.:35927500   1st Qu.:178.0  
 hcontortus_chr3_Celeg_TT_arrow_pilon:   0   Median :39185000   Median :285.0  
 hcontortus_chr4_Celeg_TT_arrow_pilon:   0   Mean   :38852688   Mean   :294.3  
 hcontortus_chr5_Celeg_TT_arrow_pilon:2008   3rd Qu.:42385000   3rd Qu.:395.0  
 hcontortus_chrX_Celeg_TT_arrow_pilon:  34   Max.   :48795000   Max.   :752.0  
                                                                               
       variable        value       
 F0.F3_I_L2:1252   Min.   :0.1842  
 F0.F3_I_L3: 649   1st Qu.:0.2072  
 F0.F3_I_L1: 145   Median :0.2344  
 F0.F1_C_L1:   0   Mean   :0.2489  
 F0.F1_C_L2:   0   3rd Qu.:0.2737  
 F0.F1_C_L3:   0   Max.   :0.5669  
 (Other)   :   0    
```
# Plotting figure for manuscript:

## Basic plot (theme as currently have)
Note, needed to reduce the text size etc to fit the tiff output file requirement sizes for PLOS Path. 
```
R

library(patchwork)

x<- ggplot(data=win_melt_F1, mapping=aes(x=1:nrow(win_melt_F1), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.8) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_I_L1","F0.F1_I_L2", "F0.F1_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_M_L1","F0.F1_M_L2", "F0.F1_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "right")+
facet_wrap(~variable.new, ncol=3, drop=FALSE)+
guides(colour = guide_legend(override.aes = list(alpha = 1, size=1.5), title = "Chromosome", legend.text.position = "left",  legend.title.position = "right", ncol = 1, legend.text = element_text(size=8), legend.title = element_text(size = 10))) +
ggtitle('F1 generation')


x2<- ggplot(data=win_melt_F2, mapping=aes(x=1:nrow(win_melt_F2), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_I_L1","F0.F2_I_L2", "F0.F2_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_M_L1","F0.F2_M_L2", "F0.F2_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE)+
guides(alpha = "none", colour = "none") +
ggtitle('F2 generation')


x3<- ggplot(data=win_melt_F3.F0, mapping=aes(x=1:nrow(win_melt_F3.F0), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_I_L1","F0.F3_I_L2", "F0.F3_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_M_L1","F0.F3_M_L2", "F0.F3_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE) +
guides(alpha = "none", colour = "none") +
ggtitle('F3 generation')

c5<- ggplot(data=subset(win_melt_selpwc, win_melt_selpwc$chrom == "Chr_5"), mapping=aes(x=mid/1e6, y=value)) +
coord_cartesian(ylim=c(0,1)) +
scale_x_continuous(
    breaks = seq(0, 55, 10),
    minor_breaks = seq(0, 55, 5)
  )+
geom_point(colour="#619CFF", alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
geom_hline(yintercept = 0.1840843, linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(yintercept = 0.2144305, linetype = "dotted", linewidth = 0.5, colour = "black") +
theme(axis.title.x = element_text(size=10), axis.text.x = element_text(size=8), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE) +
guides(alpha = "none", colour = "none") +
ggtitle('Selected vs Selected (F3 generation)')


patchwork <- x + x2 + x3 + c5 
plotme<-patchwork + plot_annotation(tag_levels = 'A')
# + plot_layout (axis_titles = "collect")
ggsave("EmergingMLR_manuscript_Fig3.tiff", plotme, device=tiff, width=2250, height=2250, units="px", dpi=320)
```
_ title = 'Haemonchus contortus Selection Lines experiment',
   caption = 'Subsampled depth = 57, Only sites with variants in all samples included, Dashed hline = F0.F3_I u+5sd,  Dotted hline = F0.F3_M u+5sd'_




## Theme_bw
Note, needed to reduce the text size etc to fit the tiff output file requirement sizes for PLOS Path. 
```
R

library(patchwork)

x<- ggplot(data=win_melt_F1, mapping=aes(x=1:nrow(win_melt_F1), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.8) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_I_L1","F0.F1_I_L2", "F0.F1_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_M_L1","F0.F1_M_L2", "F0.F1_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_bw()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "right")+
facet_wrap(~variable.new, ncol=3, drop=FALSE)+
guides(colour = guide_legend(override.aes = list(alpha = 1, size=1.5), title = "Chromosome", legend.text.position = "left",  legend.title.position = "right", ncol = 1, legend.text = element_text(size=8), legend.title = element_text(size = 10))) +
ggtitle('F1 generation')


x2<- ggplot(data=win_melt_F2, mapping=aes(x=1:nrow(win_melt_F2), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_I_L1","F0.F2_I_L2", "F0.F2_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_M_L1","F0.F2_M_L2", "F0.F2_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_bw()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE)+
guides(alpha = "none", colour = "none") +
ggtitle('F2 generation')


x3<- ggplot(data=win_melt_F3.F0, mapping=aes(x=1:nrow(win_melt_F3.F0), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_I_L1","F0.F3_I_L2", "F0.F3_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_M_L1","F0.F3_M_L2", "F0.F3_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_bw()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE) +
guides(alpha = "none", colour = "none") +
ggtitle('F3 generation')

c5<- ggplot(data=subset(win_melt_selpwc, win_melt_selpwc$chrom == "Chr_5"), mapping=aes(x=mid/1e6, y=value)) +
coord_cartesian(ylim=c(0,1)) +
scale_x_continuous(
    breaks = seq(0, 55, 10),
    minor_breaks = seq(0, 55, 5)
  )+
geom_point(colour="#619CFF", alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
geom_hline(yintercept = 0.1840843, linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(yintercept = 0.2144305, linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_bw()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_text(size=8), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE) +
guides(alpha = "none", colour = "none") +
ggtitle('Selected vs Selected (F3 generation)')


patchwork <- x + x2 + x3 + c5 
plotme<-patchwork + plot_annotation(tag_levels = 'A')

ggsave("EmergingMLR_manuscript_Fig3_bw.tiff", plotme, device=tiff, width=2250, height=2250, units="px", dpi=320)
```


## Theme_linedraw
Note, needed to reduce the text size etc to fit the tiff output file requirement sizes for PLOS Path. 
```
R

library(patchwork)

x<- ggplot(data=win_melt_F1, mapping=aes(x=1:nrow(win_melt_F1), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.8) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_I_L1","F0.F1_I_L2", "F0.F1_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_M_L1","F0.F1_M_L2", "F0.F1_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_linedraw()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "right")+
facet_wrap(~variable.new, ncol=3, drop=FALSE)+
guides(alpha = "none", colour = "none") +
ggtitle('F1 generation')


x2<- ggplot(data=win_melt_F2, mapping=aes(x=1:nrow(win_melt_F2), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_I_L1","F0.F2_I_L2", "F0.F2_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_M_L1","F0.F2_M_L2", "F0.F2_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_linedraw()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE)+
guides(alpha = "none", colour = "none") +
ggtitle('F2 generation')


x3<- ggplot(data=win_melt_F3.F0, mapping=aes(x=1:nrow(win_melt_F3.F0), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_I_L1","F0.F3_I_L2", "F0.F3_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_M_L1","F0.F3_M_L2", "F0.F3_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_linedraw()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE) +
guides(colour = guide_legend(override.aes = list(alpha = 1, size=1.5), title = "Chromosome", legend.text.position = "left",  legend.title.position = "right", nrow = 1, legend.text = element_text(size=6), legend.title = element_text(size = 8))) +
ggtitle('F3 generation')

c5<- ggplot(data=subset(win_melt_selpwc, win_melt_selpwc$chrom == "Chr_5"), mapping=aes(x=mid/1e6, y=value)) +
coord_cartesian(ylim=c(0,1)) +
scale_x_continuous(
    breaks = seq(0, 55, 10),
    minor_breaks = seq(0, 55, 5)
  )+
geom_point(colour="#619CFF", alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
geom_hline(yintercept = 0.1840843, linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(yintercept = 0.2144305, linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_linedraw()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_text(size=8), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE) +
guides(alpha = "none", colour = "none") +
ggtitle('Selected vs Selected (F3 generation)')


patchwork <- x + x2 + x3 + c5 
plotme<-patchwork + plot_annotation(tag_levels = 'A')

ggsave("EmergingMLR_manuscript_Fig3_linedraw.tiff", plotme, device=tiff, width=2250, height=2250, units="px", dpi=320)
```


## Theme_light
Note, needed to reduce the text size etc to fit the tiff output file requirement sizes for PLOS Path. 
```
R

library(patchwork)

x<- ggplot(data=win_melt_F1, mapping=aes(x=1:nrow(win_melt_F1), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.8) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_I_L1","F0.F1_I_L2", "F0.F1_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F1,variable.new==c("F0.F1_M_L1","F0.F1_M_L2", "F0.F1_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "right")+
facet_wrap(~variable.new, ncol=3, drop=FALSE)+
guides(alpha = "none", colour = "none") +
ggtitle('F1 generation')


x2<- ggplot(data=win_melt_F2, mapping=aes(x=1:nrow(win_melt_F2), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_I_L1","F0.F2_I_L2", "F0.F2_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F2, variable==c("F0.F2_M_L1","F0.F2_M_L2", "F0.F2_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE)+
guides(alpha = "none", colour = "none") +
ggtitle('F2 generation')


x3<- ggplot(data=win_melt_F3.F0, mapping=aes(x=1:nrow(win_melt_F3.F0), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Positional index")+
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_I_L1","F0.F3_I_L2", "F0.F3_I_L3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(data = filter(win_melt_F3.F0, variable==c("F0.F3_M_L1","F0.F3_M_L2", "F0.F3_M_L3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE) +
guides(colour = guide_legend(override.aes = list(alpha = 1, size=1.5), title = "Chromosome", legend.text.position = "left",  legend.title.position = "right", nrow = 1, legend.text = element_text(size=6), legend.title = element_text(size = 8))) +
ggtitle('F3 generation')

c5<- ggplot(data=subset(win_melt_selpwc, win_melt_selpwc$chrom == "Chr_5"), mapping=aes(x=mid/1e6, y=value)) +
coord_cartesian(ylim=c(0,1)) +
scale_x_continuous(
    breaks = seq(0, 55, 10),
    minor_breaks = seq(0, 55, 5)
  )+
geom_point(colour="#619CFF", alpha=0.3, size=0.6) +
labs(y="10kb window Fst value", x="Basepair (Mb)")+
geom_hline(yintercept = 0.1840843, linetype = "dashed", linewidth = 0.5, colour = "black") +
geom_hline(yintercept = 0.2144305, linetype = "dotted", linewidth = 0.5, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_text(size=8), axis.title.y = element_text(size=10), axis.text.y = element_text(size=8), strip.text.x = element_text(size = 8), plot.title = element_blank(), legend.position = "bottom")+
facet_wrap(~variable, ncol=3, drop=TRUE) +
guides(alpha = "none", colour = "none") +
ggtitle('Selected vs Selected (F3 generation)')


patchwork <- x + x2 + x3 + c5 
plotme<-patchwork + plot_annotation(tag_levels = 'A')

ggsave("EmergingMLR_manuscript_Fig3_light.tiff", plotme, device=tiff, width=2250, height=2250, units="px", dpi=320)
```
## Renaming samples for clarity:

```
# gsub https://endmemo.com/r/gsub.php
# gsub(pattern, replacement, x, ignore.case = FALSE, perl = FALSE, fixed = FALSE, useBytes = FALSE)

# CTL1, IVM1 etc:
win_melt_F1$variable.new<-gsub("F0.F1_C_L","CTL", win_melt_F1$variable.new)
win_melt_F1$variable.new<-gsub("F0.F1_I_L","IVM", win_melt_F1$variable.new)
win_melt_F1$variable.new<-gsub("F0.F1_M_L","MOX", win_melt_F1$variable.new)

win_melt_F2$variable<-gsub("F0.F2_C_L","CTL", win_melt_F2$variable)
win_melt_F2$variable<-gsub("F0.F2_I_L","IVM", win_melt_F2$variable)
win_melt_F2$variable<-gsub("F0.F2_M_L","MOX", win_melt_F2$variable)

win_melt_F3.F0$variable<-gsub("F0.F3_C_L","CTL", win_melt_F3.F0$variable)
win_melt_F3.F0$variable<-gsub("F0.F3_I_L","IVM", win_melt_F3.F0$variable)
win_melt_F3.F0$variable<-gsub("F0.F3_M_L","MOX", win_melt_F3.F0$variable)

win_melt_selpwc$variable<-gsub("F3_M_L","MOX", win_melt_selpwc$variable)
win_melt_selpwc$variable<-gsub("F3_I_L","IVM", win_melt_selpwc$variable)
win_melt_selpwc$variable<-gsub("\\.",":", win_melt_selpwc$variable)
```

## Theme_light, no legend, use grid instead - with titles
Note, needed to reduce the text size etc to fit the tiff output file requirement sizes for PLOS Path. 
```
library(patchwork)

x<- ggplot(data=win_melt_F1, mapping=aes(x=mid/1e6, y=value)) +
coord_cartesian(ylim=c(0,1)) +
scale_x_continuous(
    breaks = seq(0, 95, 20)
  )+
scale_y_continuous(
    breaks = seq(0, 1, 0.5)
  )+
geom_point(aes(colour=chrom), alpha=0.3, size=0.3) +
labs(y="Genetic differentiation (10kb window Fst)", x="Genomic position (Mb)")+
geom_hline(data = filter(win_melt_F1,variable.new==c("IVM1","IVM2", "IVM3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.3, colour = "black") +
geom_hline(data = filter(win_melt_F1,variable.new==c("MOX1","MOX2", "MOX3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.3, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_text(size=6), axis.title.y = element_text(size=10), axis.text.y = element_text(size=6), strip.text.x = element_text(size = 6, face = "bold", margin = margin(0.5,0,0.5,0, "mm")), strip.text.y = element_text(size = 6, face = "bold", angle=0), plot.title = element_text(size=10, hjust=0.5), legend.position = "none", panel.spacing.x = unit(0.2, "mm"), panel.spacing.y = unit(0.75, "mm"))+
facet_grid(variable.new~chrom, scales = "free_x", switch="x")+
guides(alpha = "none", colour = "none") +
ggtitle('F0:F1 generation')


x2<- ggplot(data=win_melt_F2, mapping=aes(x=mid/1e6, y=value)) +
coord_cartesian(ylim=c(0,1)) +
scale_x_continuous(
    breaks = seq(0, 95, 20)
  )+
scale_y_continuous(
    breaks = seq(0, 1, 0.5)
  )+
geom_point(aes(colour=chrom), alpha=0.3, size=0.3) +
labs(y="Genetic differentiation (10kb window Fst)", x="Genomic position (Mb)")+
geom_hline(data = filter(win_melt_F2, variable==c("IVM1","IVM2", "IVM3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.3, colour = "black") +
geom_hline(data = filter(win_melt_F2, variable==c("MOX1","MOX2", "MOX3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.3, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_text(size=6), axis.title.y = element_text(size=10), axis.text.y = element_text(size=6), strip.text.x = element_text(size = 6, face = "bold", margin = margin(0.5,0,0.5,0, "mm")), strip.text.y = element_text(size = 6, face = "bold", angle=0), plot.title = element_text(size=10, hjust=0.5), legend.position = "none", panel.spacing.x = unit(0.2, "mm"), panel.spacing.y = unit(0.75, "mm"))+
facet_grid(variable~chrom, scales = "free_x", switch="x")+
guides(alpha = "none", colour = "none") +
ggtitle('F0:F2 generation')


x3<- ggplot(data=win_melt_F3.F0, mapping=aes(x=mid/1e6, y=value)) +
coord_cartesian(ylim=c(0,1)) +
scale_x_continuous(
    breaks = seq(0, 95, 20)
  )+
scale_y_continuous(
    breaks = seq(0, 1, 0.5)
  )+
geom_point(aes(colour=chrom), alpha=0.3, size=0.3) +
labs(y="Genetic differentiation (10kb window Fst)", x="Genomic position (Mb)")+
geom_hline(data = filter(win_melt_F3.F0, variable==c("IVM1","IVM2", "IVM3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.3, colour = "black") +
geom_hline(data = filter(win_melt_F3.F0, variable==c("MOX1","MOX2", "MOX3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.3, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_text(size=6), axis.title.y = element_text(size=10), axis.text.y = element_text(size=6), strip.text.x = element_text(size = 6, face = "bold", margin = margin(0.5,0,0.5,0, "mm")), strip.text.y = element_text(size = 6, face = "bold", angle=0), plot.title = element_text(size=10, hjust=0.5), legend.position = "none", panel.spacing.x = unit(0.2, "mm"), panel.spacing.y = unit(0.75, "mm"))+
facet_grid(variable~chrom, scales = "free_x", switch="x") +
guides(colour = guide_legend(override.aes = list(alpha = 1, size=1.5), title = "Chromosome", legend.text.position = "left",  legend.title.position = "right", nrow = 1, legend.text = element_text(size=6), legend.title = element_text(size = 8))) +
ggtitle('F0:F3 generation')

c5<- ggplot(data=subset(win_melt_selpwc, win_melt_selpwc$chrom == "5"), mapping=aes(x=mid/1e6, y=value)) +
coord_cartesian(ylim=c(0,1)) +
scale_x_continuous(
    breaks = seq(0, 55, 10),
    minor_breaks = seq(0, 55, 5)
  )+
scale_y_continuous(
    breaks = seq(0, 1, 0.5)
  )+
geom_point(colour="#619CFF", alpha=0.3, size=0.3) +
labs(y="Genetic differentiation (10kb window Fst)", x="Genomic position (Mb)")+
geom_hline(yintercept = 0.1840843, linetype = "dashed", linewidth = 0.3, colour = "black") +
geom_hline(yintercept = 0.2144305, linetype = "dotted", linewidth = 0.3, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_text(size=6), axis.title.y = element_text(size=10), axis.text.y = element_text(size=6), strip.text.x = element_text(size = 6, face = "bold", margin = margin(0.5,0,0.5,0, "mm")), strip.text.y = element_text(size = 6, face = "bold", angle=0), plot.title = element_text(size=10, hjust=0.5), legend.position = "none", panel.spacing.x = unit(0.2, "mm"), panel.spacing.y = unit(0.75, "mm"))+
facet_grid(variable~chrom, scales = "free_x", switch="x") +
guides(alpha = "none", colour = "none") +
ggtitle('F3 generation: Ivermectin vs Moxidectin lines')


patchwork <- x + x2 + x3 + c5  
plotme<-patchwork + plot_annotation(tag_levels = 'A') + plot_layout(axis_titles = "collect")

ggsave("EmergingMLR_manuscript_Fig3_light_titles.tiff", plotme, device=tiff, width=2250, height=2250, units="px", dpi=320)
```
![image](https://github.com/user-attachments/assets/3d0650fb-2b73-4111-8890-4e59b228e465)


## Theme_light, no legend, use grid instead - with titles - BUT WITHOUT faceting by chromosome
Note, needed to reduce the text size etc to fit the tiff output file requirement sizes for PLOS Path. 
```
# Note that you need to make a new variable as have renamed variable.new, and make it a factor again (got changed to a character when renamed)
win_melt_F1$variable.new2 <- factor(win_melt_F1$variable.new, levels=(c("CTL1", "CTL2", "CTL3", "", "IVM2", "IVM3", "MOX1", "MOX2", "MOX3")))   


library(patchwork)

# Note that to get it to plot (not sure why) you need to include "IVM1" for the hline, despite this not being present. Yet you don't have to have "MOX1" despite that being present... 
x<- ggplot(data=win_melt_F1, mapping=aes(x=1:nrow(win_melt_F1), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.3) +
labs(y="Genetic differentiation (10kb window Fst)", x="Positional Index")+
geom_hline(data = filter(win_melt_F1,variable.new2==c("IVM1","IVM2", "IVM3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.3, colour = "black") +
geom_hline(data = filter(win_melt_F1,variable.new2==c("MOX1","MOX2", "MOX3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.3, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=6), strip.text.x = element_text(size = 6, face = "bold", margin = margin(0.5,0,0.5,0, "mm")), strip.text.y = element_text(size = 6, face = "bold", angle=0), plot.title = element_text(size=10, hjust=0.5), legend.position = "none", panel.spacing.x = unit(0.2, "mm"), panel.spacing.y = unit(0.75, "mm"))+
facet_wrap(~variable.new2, ncol=3, drop=FALSE)+
guides(alpha = "none", colour = "none") +
ggtitle('F0:F1 generation')


x2<- ggplot(data=win_melt_F2, mapping=aes(x=1:nrow(win_melt_F2), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.3) +
labs(y="Genetic differentiation (10kb window Fst)", x="Positional Index")+
geom_hline(data = filter(win_melt_F2, variable==c("IVM1","IVM2", "IVM3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.3, colour = "black") +
geom_hline(data = filter(win_melt_F2, variable==c("MOX1","MOX2", "MOX3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.3, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=6), strip.text.x = element_text(size = 6, face = "bold", margin = margin(0.5,0,0.5,0, "mm")), strip.text.y = element_text(size = 6, face = "bold", angle=0), plot.title = element_text(size=10, hjust=0.5), legend.position = "none", panel.spacing.x = unit(0.2, "mm"), panel.spacing.y = unit(0.75, "mm"))+
facet_wrap(~variable, ncol=3, drop=TRUE)+
guides(alpha = "none", colour = "none") +
ggtitle('F0:F2 generation')


x3<- ggplot(data=win_melt_F3.F0, mapping=aes(x=1:nrow(win_melt_F3.F0), y=value)) +
coord_cartesian(ylim=c(0,1)) +
geom_point(aes(colour=chrom), alpha=0.3, size=0.3) +
labs(y="Genetic differentiation (10kb window Fst)", x="Positional Index")+
geom_hline(data = filter(win_melt_F3.F0, variable==c("IVM1","IVM2", "IVM3")), aes(yintercept = 0.1840843), linetype = "dashed", linewidth = 0.3, colour = "black") +
geom_hline(data = filter(win_melt_F3.F0, variable==c("MOX1","MOX2", "MOX3")), aes(yintercept = 0.2144305), linetype = "dotted", linewidth = 0.3, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_blank(), axis.title.y = element_text(size=10), axis.text.y = element_text(size=6), strip.text.x = element_text(size = 6, face = "bold", margin = margin(0.5,0,0.5,0, "mm")), strip.text.y = element_text(size = 6, face = "bold", angle=0), plot.title = element_text(size=10, hjust=0.5), legend.position = "bottom", panel.spacing.x = unit(0.2, "mm"), panel.spacing.y = unit(0.75, "mm"))+
facet_wrap(~variable, ncol=3, drop=TRUE)+
guides(colour = guide_legend(override.aes = list(alpha = 1, size=1.5), title = "Chromosome", legend.text.position = "left",  legend.title.position = "top", nrow = 1, legend.text = element_text(size=6), legend.title = element_text(size = 8))) +
ggtitle('F0:F3 generation')

c5<- ggplot(data=subset(win_melt_selpwc, win_melt_selpwc$chrom == "5"), mapping=aes(x=mid/1e6, y=value)) +
coord_cartesian(ylim=c(0,1)) +
scale_x_continuous(
    breaks = seq(0, 55, 10),
    minor_breaks = seq(0, 55, 5)
  )+
scale_y_continuous(
    breaks = seq(0, 1, 0.5)
  )+
geom_point(colour="#619CFF", alpha=0.3, size=0.3) +
labs(y="Genetic differentiation (10kb window Fst)", x="Genomic Position (Mb)")+
geom_hline(yintercept = 0.1840843, linetype = "dashed", linewidth = 0.3, colour = "black") +
geom_hline(yintercept = 0.2144305, linetype = "dotted", linewidth = 0.3, colour = "black") +
theme_light()+
theme(axis.title.x = element_text(size=10), axis.text.x = element_text(size=6), axis.title.y = element_text(size=10), axis.text.y = element_text(size=6), strip.text.x = element_text(size = 6, face = "bold", margin = margin(0.5,0,0.5,0, "mm")), strip.text.y = element_text(size = 6, face = "bold", angle=0), plot.title = element_text(size=10, hjust=0.5), legend.position = "none", panel.spacing.x = unit(0.2, "mm"), panel.spacing.y = unit(0.75, "mm"))+
facet_wrap(~variable, ncol=3, drop=TRUE)+
guides(alpha = "none", colour = "none") +
ggtitle('F3 generation: Ivermectin vs Moxidectin lines')


patchwork <- x + x2 + x3 + c5  
plotme<-patchwork + plot_annotation(tag_levels = 'A') + plot_layout(axis_titles = "collect_y")

ggsave("EmergingMLR_manuscript_Fig3_light_titles_positionalindex.tiff", plotme, device=tiff, width=2250, height=2250, units="px", dpi=320)
```
![image](https://github.com/user-attachments/assets/ae3864c5-6afe-4965-b06b-6c8b5ec91eca)

