This will produce the mpileup file, from which you can then make pileup files

You use the mpileup file for PoPoolation2 and Grenedalf, and the pileup files for PoPoolation (1)

1. First, you want to create a list of all your bam files that contain data you want in the mpileup file. Note that you want it to be in the order you desire your samples to be in in your mpileup.

All completed bwa mem aligned files are here: ```/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads/completed```, and are within their own bwamem_splitter output directories. 

```
find "$PWD" -name \*merged.sorted.marked.bam > bam.list
sort bam.list > sbam.list
```
This has not sorted it as I want it to (donor numbers etc not helping). Opened in Excel and manually sorted it.

```
/home/jenni/data_folder/selection_lines_poolseq_bams/16-11098_F0.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/13-13804_postCTL1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/14-13775_postCTL1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/10-13796_postCTL1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/1-13981_CTL2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/7-13989_postCTL2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/4-13844_postCTL2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/1-14031_CTL3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/3-14949_postCTL3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/2-14036_postCTL3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/12-13788_postIVM1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/15-13766_postIVM1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/3-13982_postIVM2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/9-14008_postIVM2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/6-13911_postIVM2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/5-14916_postIVM3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/4-14907_postIVM3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/6-14917_postIVM3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/11-13740_postMOX1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/13-13807_postMOX1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/14-13723_postMOX1.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/2-13934_postMOX2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/5-13918_postMOX2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/8-13930_postMOX2.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/7-14029_postMOX3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/9-14954_postMOX3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/8-14953_postMOX3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/12-Kaplan_TxPx2011.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/17-Kaplan_UGA2020.merged.sorted.marked.bam
```
Made the above into a new bam.list file. 

In the working directory (/home/jenni/data_folder/working_folder/selection_lines_poolseq/):
```
mkdir mpileup
cd mpileup

ln -s /home/jenni/data_folder/selection_lines_poolseq_bams/bam.list bam.list
```

2. Next, obtain your reference genome (**MHco3(ISE) WBPS18 HCON_V4**) (can use wget <ref.fa>). You then need to index it, which makes it a lot faster for programs to run (and is required).

```
wget https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS18/species/haemonchus_contortus/PRJEB506/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa.gz

gunzip haemonchus_contortus.PRJEB506.WBPS18.genomic.fa.gz
```
Put symbolic link in the mpileup directory.
```
ln -s /home/jenni/data_folder/working_folder/selection_lines_poolseq/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa  ref.fa
```

To index the reference genome note that it must be a .fa not .fa.gz file!

```
module load samtools/1.17

samtools --version > samtools_version_index

samtools faidx ref.fa
samtools dict ref.fa > ref.dict
```

3. To create an mpileup file:

```
samtools --version > samtools_version_mpileup

samtools mpileup -B -q 20 -Q 30 --ff DUP -f ref.fa -b bam.list > Sel_lines_poolseq_Hcwbps18_q20Q30.mpileup
```
The file is 1.5Tb in size!

To check it is complete, looked at the tail of the .fai file and cf last contig in the mpileup - all good.
Also:
``` 
wc -l Sel_lines_poolseq_Hcwbps18_q20Q30.mpileup
# 275364588 Sel_lines_poolseq_Hcwbps18_q20Q30.mpileup
# 275364588 Sel_lines_poolseq_Hcwbps18_q20Q30.mpileup2
```

```
http://www.htslib.org/doc/samtools-mpileup.html (v1.17)

-q, --min-MQ INT
Minimum mapping quality for an alignment to be used [0]

-Q, --min-BQ INT
Minimum base quality for a base to be considered. [13]

Note base-quality 0 is used as a filtering mechanism for overlap removal which marks bases as having quality zero and lets the base quality filter remove them. Hence using --min-BQ 0 will make the overlapping bases reappear, albeit with quality zero.

```


4. To change the file format for the next step (for PoPoolation2) - this checks that there are no empty colums (made as two tabs by error):

Note, submitting it on James' server, so will use ```nohup``` to make it continue running and produce log files after I logout the ssh

```
nohup - so that it will not hang up and stop when the shell terminates (log out/internet dies/computer dies) but better than disown as you can still see the job next time you log in I think
2>error.log redirects error messages to a second file (otherwise they get dumped with your actual output in the main output file). 
& backgrounds the process so you can continue working on other stuff
Note that if you don't give it an output file it will put eveyrthing in nohup.out, and it will do this for all your jobs! 
```
```
nohup sed 's/\t\t/\t!\t!/g' Sel_lines_poolseq_Hcwbps18_q20Q30.mpileup > Sel_lines_poolseq_Hcwbps18_q20Q30.mpileup2 2>convert_mpileup_error.log &
```

I installed popoolation2_1201 (just unzip)

5. Then let's ID the indels:
```
nohup perl /home/jenni/data_folder/software/popoolation2_1201/indel_filtering/identify-indel-regions.pl --indel-window 5 --min-count 2 --input Sel_lines_poolseq_Hcwbps18_q20Q30.mpileup2 --output Sel_lines_poolseq_Hcwbps18_q20Q30.indels.gtf 2>indel_id_error.log &
```
**Output:**

Pileup entries processed: 275,364,588

Pileup entries containing at least one indel: 20,460,500

How many bp of the reference are covered by indel-regions: 65,379,005

6. Then sync the file:

```
nohup java -ea -Xmx7g -jar /home/jenni/data_folder/software/popoolation2_1201/mpileup2sync.jar --input Sel_lines_poolseq_Hcwbps18_q20Q30.mpileup2 --output Sel_lines_poolseq_Hcwbps18_q20Q30.java.sync --fastq-type sanger --min-qual 20 --threads 8 &
```

```
275,364,588 Sel_lines_poolseq_Hcwbps18_q20Q30.java.sync`(wc -l)
```
Note, output is:
```
column >3: allele frequencies for all populations in the form A-count:T-count:C-count:G-count:N-count:deletion-count
```
7. And filter the indels:
```
nohup perl /home/jenni/data_folder/software/popoolation2_1201/indel_filtering/filter-sync-by-gtf.pl --input Sel_lines_poolseq_Hcwbps18_q20Q30.java.sync --gtf Sel_lines_poolseq_Hcwbps18_q20Q30.indels.gtf --output Sel_lines_poolseq_Hcwbps18_q20Q30_noindels.java.sync 2>filter_indel_error.log &
```
