I am re-subsampling the mpileup file, using only the selection line samples (as it considers a site to be included or excluded using all samples, which can exclude more than I want)

Will subsample to a coverage depth of 57 - allows potentially 75% of all sites to be included. 

First, I need to re-order the sync mpileup, as I have only re-ordered subsampled ones before:
```
First, rearrange the mpileup so that all the samples are in line order. This will make life so much easier! Note that I first went back through logs and directories on the terminal. I began re-arranging files into the right order AFTER the 4th August. Therefore, the sync file will not have been changed (July). It was also made on the 11th July, and indels were identifed on the 10th. So definitely the original!

awk 'BEGIN{OFS = "\t"}{print $1,$2,$3,$4,$7,$6,$5,$8,$10,$9,$12,$11,$13,$14,$15,$16,$18,$17,$21,$19,$20,$22,$23,$24,$25,$26,$27,$29,$28,$30,$31,$32}' Sel_lines_poolseq_Hcwbps18_q20Q30_noindels.java.sync > tmp &

mv tmp Sel_lines_poolseq_Hcwbps18_q20Q30_noindel_lines_in_correct_order_Jan24.java.sync
```
```
# Get just the SL (not Kaplan):

cat Sel_lines_poolseq_Hcwbps18_q20Q30_noindel_lines_in_correct_order_Jan24.java.sync | awk '{print NF}'

# 32 - so remove the last 2 (KaplanS and R)

cut -f -30 Sel_lines_poolseq_Hcwbps18_q20Q30_noindel_lines_in_correct_order_Jan24.java.sync > SL_ps_only_Hcwbps18_q20Q30_noindel_lines_in_correct_order_Jan24.java.sync
```

```
# Subsample to a uniform coverage across all samples
perl /home/jenni/data_folder/software/popoolation2_1201/subsample-synchronized.pl --input /home/jenni/data_folder/working_folder/selection_lines_poolseq/mpileup/SL_ps_only_Hcwbps18_q20Q30_noindel_lines_in_correct_order_Jan24.java.sync --output SLps_only_co_Hcwbps18_q20Q30_noindels_ss57.java.sync --target-coverage 57 --max-coverage 2%  --method withoutreplace 2>subsample_error.log &


disown -h jobid
```
```
# From the perl script:

=item B<--target-coverage>

Reduce the coverage of the pileup-file to the here provided value; The target coverage also acts as minimum coverage,
i.e.: if the coverage in any population is smaller than the targetcoverage the whole pileup entry is discarded. Mandatory

=item B<--max-coverage>

The maximum coverage; All populations are required to have coverages lower or equal than the maximum coverage; Mandatory
The maximum coverage may be provided as one of the following:

 '500' a maximum coverage of 500 will be used for all populations
 '300,400,500' a maximum coverage of 300 will be used for the first population, a maximum coverage of 400 for the second population and so on
 '2%' the 2% highest coverages will be ignored, this value is independently estimated for every population

=item B<--method>

Specify the method for subsampling of the synchronized file. Either: withreplace, withoutreplace, fraction; Mandatory
```

Checked had completed ok - error file empty and had all six chromosomes represented in the file.

Total number of sites: 
```
wc -l
# 89952741
```
So, have covered a total of:

283,439,308 = total genome length

Have inlcuded more sites than when had the Kaplan samples. 31.7% 
