
First download the genome.
```
#!/bin/sh
#PBS -l walltime=2:00:00
#PBS -l cput=2:00:00
#PBS -l nodes=1:centos7
#PBS -d /export/home2/jmi45g/lainglab/GENOMES

wget https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS18/species/haemonchus_contortus/PRJEB506/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa.gz

```
Obtain the pre-trimmed sequencing files:
```
mkdir Fastq_files/renamed_fastq_files
mv Trimmed/Sample*/*R[1-2].fastq.gz Fastq_files/renamed_fastq_files/
```
**Correct fastq headers so that barcodes included are in samtools friendly format!**

This was neccessary for these files, but will not be the case for all files. 

Note, will run in groups of files for speed, given how long the fastQC took. Will write as run them. To start with those numbered 1-*.
```
#!/bin/sh 
#PBS -l walltime=50:00:00 
#PBS -l cput=6:00:00 
#PBS -l nodes=1:centos7 
#PBS -d /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/

#for i in 1-*R[1-2].fastq.gz; do sed 's/1:N:0/BC:Z/g' $i | sed 's/2:N:0/BC:Z/g' > ${i}_renamed ; done  # Note, this output the new files, but headers were unchanged. 
for i in 1-*R[1-2].fastq.gz; do zcat $i | sed 's/1:N:0/BC:Z/g' | sed 's/2:N:0/BC:Z/g' > ${i}_renamed ; done
```
Check headers using:
```
zcat 1-13981_postCTL2_L002_R1.fastq.gz | head -n 10
```
Rename files for bwamem splitter script:
```
rename .gz_renamed .gz *_renamed # Rename all files - command is 'rename from to file_name'
```
Then, align the reads using the bwamem_splitter script adapted from one originally written by Dr Stephen Doyle.
```
mkdir Align_reads

# The BWA-MEM splitter script does not work in one go as on the WSI. 
# You need to run several different jobs. 

qsub ./run_bwamem_Poolseq_2023.sh -v VAR1="1-13981_CTL2",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/1-13981_postCTL2_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/1-13981_postCTL2_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="1-14031_CTL3",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/1-14031_postCTL3_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/1-14031_postCTL3_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="2-13934_postMOX2",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/2-13934_postMOX2_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/2-13934_postMOX2_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="2-14036_postCTL3",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/2-14036_postCTL3_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/2-14036_postCTL3_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="3-13982_postIVM2",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/3-13982_postIVM2_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/3-13982_postIVM2_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="3-14949_postCTL3",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/3-14949_postCTL3_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/3-14949_postCTL3_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="4-13844_postCTL2",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/4-13844_postCTL2_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/4-13844_postCTL2_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="4-14907_postIVM3",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/4-14907_postIVM3_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/4-14907_postIVM3_L001_R2.fastq.gz"

# Note, from this point onwards, should output the step 2.2 logfiles to my ~/logfiles directory (to stop the error emails!)
qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="5-13918_postMOX2",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/5-13918_postMOX2_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/5-13918_postMOX2_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="5-14916_postIVM3",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/5-14916_postIVM3_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/5-14916_postIVM3_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="6-13911_postIVM2",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/6-13911_postIVM2_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/6-13911_postIVM2_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="6-14917_postIVM3",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/6-14917_postIVM3_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/6-14917_postIVM3_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="7-13989_postCTL2",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/7-13989_postCTL2_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/7-13989_postCTL2_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="7-14029_postMOX3",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/7-14029_postMOX3_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/7-14029_postMOX3_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="8-13930_postMOX2",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/8-13930_postMOX2_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/8-13930_postMOX2_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="8-14953_postMOX3",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/8-14953_postMOX3_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/8-14953_postMOX3_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="9-14008_postIVM2",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/9-14008_postIVM2_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/9-14008_postIVM2_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="9-14954_postMOX3",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/9-14954_postMOX3_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/9-14954_postMOX3_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="10-13796_postCTL1",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/10-13796_CTL1_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/10-13796_CTL1_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="11-13740_postMOX1",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/11-13740_MOX1_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/11-13740_MOX1_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="12-13788_postIVM1",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/12-13788_postIVM1_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/12-13788_postIVM1_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="13-13804_postCTL1",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/13-13804_postCTL1_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/13-13804_postCTL1_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="13-13807_postMOX1",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/13-13807_postMOX1_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/13-13807_postMOX1_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="14-13723_postMOX1",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/14-13723_postMOX1_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/14-13723_postMOX1_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="14-13775_postCTL1",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/14-13775_postCTL1_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/14-13775_postCTL1_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="15-13766_postIVM1",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/15-13766_postIVM1_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/15-13766_postIVM1_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="16-11098_F0",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/16-11098_F0_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/16-11098_F0_L002_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="12-Kaplan_TxPx2011",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/12-Kaplan_TxPx2011_L001_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/12-Kaplan_TxPx2011_L001_R2.fastq.gz"

qsub ./run_bwamem_Poolseq_2023_centos7.sh -v VAR1="17-Kaplan_UGA2020",VAR2="/export/home2/jmi45g/lainglab/GENOMES/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa",VAR3="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/17-Kaplan_UGA2020_L002_R1.fastq.gz",VAR4="/export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Fastq_files/renamed_fastq_files/17-Kaplan_UGA2020_L002_R2.fastq.gz"

# This will run the script, producing the new scripts within it and running step1 and step2.1
# It will then stop. 
```
Note on script (note that the online manual has fewer commands than the software manual):

-Y      Use soft clipping for supplementary alignments

-M	Mark shorter split hits as secondary (for Picard compatibility).

-C	Append append FASTA/Q comment to SAM output. This option can be used to transfer read meta information (e.g. barcode) to the SAM output. Note that the FASTA/Q comment (the string after a space in the header line) must conform the SAM spec (e.g. BC:Z:CGTAC). Malformated comments lead to incorrect SAM output.


_Note - in samtools stats, the 'average quality' is NOT the mapq! 'average quality - ratio between the sum of base qualities and total length.'_

# Correction/change to make as on lainglab:
Cannot have ln -s as a command. Therefore, want to point directly to the reference. This will then run, but it will run in the reference folder. This may not be an issue. Will have to check through the commands. Also need to point directly to the fastq files etc. And then ditto - see if it will make an issue. 
Or - will need to cp the file to the folder (this IS possible). This may work better I think. 


```
##########################################################################################################################################################################
######################################## run_bwamem_DNAzoo_May2021 #######################################################################################################
##########################################################################################################################################################################

#!/bin/sh
#PBS -l walltime=50:00:00
#PBS -l cput=100:00:00
#PBS -l nodes=1:centos7
#PBS -d /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads
#PBS -o /export/home2/jmi45g
#PBS -e /export/home2/jmi45g

source ~/.bashrc

# BWA-MEM alignment of reads shell script for Pool-Seq:

#!/usr/bin/env bash
############################################################################################
# run_bwamem_splitter.sh
#
# sd21@sanger.ac.uk July 2018
#
# Adjusted for Glasgow HPCC by Jennifer McIntyre August 2020 jennifer.mcintyre@glasgow.ac.uk
#############################################################################################
sample_name=$VAR1
reference=$VAR2
read1=$VAR3
read2=$VAR4

ID="U$(date +%s)"

#if [ "$#" -eq 0 ] || [ "$1" == "-h" ] || [ "$1" == "--help" ]; then
#        echo ""
#    echo "Usage: ~sd21/bash_scripts/run_bwa_splitter <SAMPLE_PREFIX> <REFERENCE> <R1.fastq> <R2.fastq>"
#    echo ""
#    exit 0
#fi

if [ -d "${sample_name}_bwasplitter_out" ]; then
        echo -e "\nThere is already a run started with this sample name. Rename and start again\n"
    exit 0
fi

mkdir ${sample_name}_bwasplitter_out
cd ${sample_name}_bwasplitter_out
mkdir logfiles

# prepare reference and split data
echo -e "# prepare reference and split the raw data

#!/bin/sh
#PBS -l walltime=100:00:00
#PBS -l cput=100:00:00
#PBS -l nodes=1:centos7
#PBS -d /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads/${sample_name}_bwasplitter_out
#PBS -o /export/home2/jmi45g
#PBS -e /export/home2/jmi45g

export PATH="/export/home2/jmi45g/anaconda3/envs/aligners/bin:$PATH"

sample_name=$VAR1
reference=$VAR2
read1=$VAR3
read2=$VAR4

#ln -sf $"{reference}" ref.fa
cp $"{reference}" ref.fa
bwa index -b 100000000 ref.fa

#if [[ $read1 =~ \.gz$ ]]
#then ln -sf $"{read1}" R1.fq.gz; zcat R1.fq.gz | split -d -a 3 -l 4000000 - R1_tmp_
#else ln -sf $"{read1}" R1.fq; split -d -a 3 -l 4000000 R1.fq R1_tmp_
#fi

if [[ $read1 =~ \.gz$ ]]
then cp $"{read1}" R1.fq.gz; zcat R1.fq.gz | split -d -a 3 -l 4000000 - R1_tmp_
else cp $"{read1}" R1.fq; split -d -a 3 -l 4000000 R1.fq R1_tmp_
fi

#if [[ $read2 =~ \.gz$ ]]
#then ln -sf $"{read2}" R2.fq.gz; zcat R2.fq.gz | split -d -a 3 -l 4000000 - R2_tmp_
#else ln -sf $"{read2}" R2.fq; split -d -a 3 -l 4000000 R2.fq R2_tmp_
#fi

if [[ $read2 =~ \.gz$ ]]
then cp $"{read2}" R2.fq.gz; zcat R2.fq.gz | split -d -a 3 -l 4000000 - R2_tmp_
else cp $"{read2}" R2.fq; split -d -a 3 -l 4000000 R2.fq R2_tmp_
fi

#split -d -a 3 -l 4000000 R1.fq R1_tmp_
#split -d -a 3 -l 4000000 R2.fq R2_tmp_
touch step1_FINISHED" > step1_bwasplitter
#chmod a+x step1_bwasplitter

# prepare split mapping and set off mapping
echo -e "# prepare the split mapping run files

#!/bin/sh
#PBS -l walltime=50:00:00
#PBS -l cput=80:00:00
#PBS -l nodes=1:centos7
#PBS -d /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads/${sample_name}_bwasplitter_out
#PBS -o /export/home2/jmi45g
#PBS -e /export/home2/jmi45g

export PATH="/export/home2/jmi45g/anaconda3/envs/aligners/bin:$PATH"
export PATH="/export/home2/jmi45g/anaconda3/envs/picard_suite/bin:$PATH"

sample_name=$VAR1
n=0
for i in \` ls -1 R1_* \` ; do
let \"n+=1\"
echo -e \"
#!/bin/sh
#PBS -l walltime=50:00:00
#PBS -l cput=100:00:00
#PBS -l nodes=1:centos7
#PBS -d /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads/${sample_name}_bwasplitter_out
#PBS -o /export/home2/jmi45g
#PBS -e /export/home2/jmi45g

export PATH="/export/home2/jmi45g/anaconda3/envs/picard_suite/bin:$PATH"

/export/home2/jmi45g/anaconda3/envs/aligners/bin/bwa mem -t 4 -R '@RG\\\\\\\\\\\tRG:$"{sample_name}"\\\\\\\\\\\tID:$"{sample_name}"\\\\\\\\\\\tSM:$"{sample_name}"' -Y -M -C ref.fa $"{i}" $"{i/R1/R2}" | samtools view --threads 4 -b - | samtools sort --threads 4 -o $"{i/R1/bwamem}".tmp.sort.bam - \" > step2.2_bwamem_tmp_$"{n}"; done; #chmod a+x step2.2_bwamem_tmp_*
touch step2_FINISHED" > step2.1_bwasplitter

#chmod a+x step2.1_bwasplitter

# merge mapping, mark duplicates, generate stats, and finalise

echo -e "# merge the bam files
#!/bin/sh
#PBS -l walltime=50:00:00
#PBS -l cput=100:00:00
#PBS -l nodes=1:centos7
#PBS -d /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads/${sample_name}_bwasplitter_out
#PBS -o /export/home2/jmi45g
#PBS -e /export/home2/jmi45g

export PATH="/export/home2/jmi45g/anaconda3/envs/picard_suite/bin:$PATH"

ls -1 *.tmp.sort.bam > bam.fofn
samtools merge --threads 4 -cpf -b bam.fofn tmp.merged.sorted.bam
#rm *.tmp.sort.bam" > step3_merge

#chmod +x step3_merge


echo -e "# mark duplicates, generate stats, and finalise


#!/bin/sh
cd $PWD

export PATH="/export/home2/jmi45g/anaconda3/envs/picard_suite/bin:$PATH"

sample_name=$VAR1


ulimit -Sn > snlimit.txt # This will show you the number of jobs you have access to as a check

java -Xmx8g -jar /export/home2/jmi45g/anaconda3/envs/picard_suite/share/picard-2.5.0-2/picard.jar MarkDuplicates INPUT=tmp.merged.sorted.bam OUTPUT=tmp.merged.sorted.marked.bam METRICS_FILE=tmp.merged.sorted.marked.metrics TMP_DIR=$PWD/tmp
samtools flagstat tmp.merged.sorted.marked.bam > $"{sample_name}".merged.sorted.marked.flagstat
samtools stats tmp.merged.sorted.marked.bam | grep ^SN | cut -f 2- > $"{sample_name}".merged.sorted.marked.stats
bamtools stats -in tmp.merged.sorted.marked.bam > $"{sample_name}".merged.sorted.marked.bamstats
samtools view --threads 4 -F 12 -b tmp.merged.sorted.marked.bam -o $"{sample_name}".merged.sorted.marked.bam
samtools view --threads 4 -f 12 tmp.merged.sorted.marked.bam -o $"{sample_name}".unmapped.bam
samtools index -b $"{sample_name}".merged.sorted.marked.bam
#rm R[12]_*
#rm *.tmp.*
#mv *.[eo] logfiles/
touch step3_FINISHED
touch bam_splitter_COMPLETE" > step3_bwasplitter_centos7-run
#chmod a+x step3_bwasplitter_centos7-run

# Make a centos-7 master script to run script step3_bwasplitter_centos7-run

echo -e " # make a centos-7 master script to run script step3_bwasplitter_centos7-run
#!/bin/sh
#PBS -l walltime=50:00:00
#PBS -l cput=100:00:00
#PBS -l nodes=1:centos7
#PBS -d /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads/${sample_name}_bwasplitter_out
#PBS -o /export/home2/jmi45g
#PBS -e /export/home2/jmi45g

ssh `hostname` /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads/${sample_name}_bwasplitter_out/step3_bwasplitter_centos7-run" > step3_bwasplitter_centos7-master
#chmod a+x step3_bwasplitter_centos7-master

#### NOTE ##### The centos7 master script will be made BUT it will put in the headnode etc instead of hostname.
#### Once made, change back to hostname to run. 

#----- RELEASE THE KRAKEN!
# run - reference and paired read setup
./step1_bwasplitter -e step1_bwasplitter.e -o step1_bwasplitter.o -v VAR1="${sample_name}",VAR2="${reference}",VAR3="${read1}",VAR4="${read2}"

# run - prepare mapping scripts
./step2.1_bwasplitter -e step2.1_bwasplitter.e -o step2.1_bwasplitter.o -v VAR1="${sample_name}",VAR2="${reference}",VAR3="${read1}",VAR4="${read2}"

while [ ! -f step1_FINISHED ]
do
  sleep 2
done

jobs=$( ls -1 R1_tmp_* | wc -l )
```

08.05.21
Time take - overnight run to finish first part (for all 6, run separately)

```
####################### To set step2.2 jobs running all at once #######################################

# To try to submit multiple jobs at the same time do: 
# Note that you will want to adjust the upper limit to be more than the total number of step2.2 jobs that you have 
# Adding 'sleep 1' adds a pause of 1 second between each step, to avoid overloading HPCC

# for i in {0..9}; do qsub pbs.script.$i ; done
for i in {0..300}; do qsub step2.2_bwamem_tmp_$i ; sleep 1 ; done

# Set running in each sample folder.
```
```
###################### To set step3 running ###########################################################
# Once step2.2 jobs have ALL completed.

# First make a merged bam file: 

#!/bin/sh
#PBS -l walltime=50:00:00
#PBS -l cput=100:00:00
#PBS -l nodes=1:centos7
#PBS -d /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads/ERR3467689_bwasplitter_out

export PATH="/export/home2/jmi45g/anaconda3/envs/latest_samtools/bin:$PATH"
# This will be samtools v1.12 on 08.05.21

ls -1 *.tmp.sort.bam > bam.fofn
samtools merge --threads 4 -cpf -b bam.fofn tmp.merged.sorted.bam
```

Note that you need to be able to open LOTS of jobs on the HPCC all at once! 
Mark changed something allowing me more access to many areas and set it running with a script within a script. 
It worked! The script is below. The new snlimit was 16384.

You need to run the master script below, which you will need to write manually, after first making the step3_bwasplitter_centos7-run script via the bwamem script (see below).
```
#######-bash-3.2$  more step3_bwasplitter_centos7-master ##########
#!/bin/sh
#PBS -l walltime=50:00:00
#PBS -l cput=100:00:00
#PBS -l nodes=1:centos7
#PBS -d /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads/ERR3467689_bwasplitter_out

ssh `hostname` /export/home2/jmi45g/lainglab/HCON/Selection_lines_poolseq/Align_reads/ERR3467689_bwasplitter_out/step3_bwasplitter_centos7-run
```

**Versions used (May 2021 to April 2022):
BWA MEM - 0.7.17-r1188
Samtools - 1.3
Picard - 2.5.0-2
Bamtools - 2.5.1 

Note, for reference, this is what picard did: 
```
[Wed May 24 12:31:39 BST 2023] picard.sam.markduplicates.MarkDuplicates INPUT=[tmp.merged.sorted.bam] OUTPUT=tmp.merged.sorted.marked.bam METRICS_FILE=tmp.merged.sorted
.marked.metrics TMP_DIR=[/export/lainglab/HCON/Selection_lines_poolseq/Align_reads/4-14907_postIVM3_bwasplitter_out/tmp]    MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP=50000 M
AX_FILE_HANDLES_FOR_READ_ENDS_MAP=8000 SORTING_COLLECTION_SIZE_RATIO=0.25 REMOVE_SEQUENCING_DUPLICATES=false TAGGING_POLICY=DontTag REMOVE_DUPLICATES=false ASSUME_SORTE
D=false DUPLICATE_SCORING_STRATEGY=SUM_OF_BASE_QUALITIES PROGRAM_RECORD_ID=MarkDuplicates PROGRAM_GROUP_NAME=MarkDuplicates READ_NAME_REGEX=<optimized capture of last t
hree ':' separated fields as numeric values> OPTICAL_DUPLICATE_PIXEL_DISTANCE=100 VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORD
S_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json
```
