To run SnpEff (Cingolani et al., 2012) to find out whether any common mutations (SNPs or indels) are potentially driving the DEG seen over the locus in the selected lines.

Download and install snpEff ()

```
cd ~/data_folder/software

# Download latest version
wget https://snpeff.blob.core.windows.net/versions/snpEff_latest_core.zip

# Unzip file
unzip snpEff_latest_core.zip
```
Version installed = SnpEff version SnpEff 4.1k (build 2015-09-07), but the package was updated on the 21st June 2023. So I am assuming that is all fine! Have tried multiple different ways to download, and all say the same version. 

Will focus on F3 generation.

# Build database
First you have to make a new genome database, as H. contortus does not have one available. 

https://pcingola.github.io/SnpEff/snpeff/build_db/

1. Edit the config file to create the new genome:

For help using vi see: https://www.cs.colostate.edu/helpdocs/vi.html
```
vi snpEff.config
```
Add the following lines (you are editing snpEff.config):
```
# Mouse genome, version mm37.61
mm37.61.genome : Mouse
```
```
# Haemonchus contortus genome, version Hcwbps18
Hcwbps18.genome : Haemonchus_contortus
        Hcwbps18.mitochondrion.codonTable : Invertebrate_Mitochondrial

```
 2. Create directory for this new genome
```
mkdir data # within snpEff
cd snpEff/data
mkdir Hcwbps18
cd Hcwbps18
```

3. Get annotation file â€“ create a link to the .gtf file. Call this genes.gff
```
ln -s /home/jenni/data_folder/wbps18/haemonchus_contortus.PRJEB506.WBPS18.annotations.gtf genes.gtf
```
Create a link to the genome (note, this needs to be in the 'genomes' folder). 
Call this Hcwbps18.fa
```
mkdir snpEff/data/genomes
cd snpEff/data/genomes
ln -s /home/jenni/data_folder/wbps18/haemonchus_contortus.PRJEB506.WBPS18.genomic.fa Hcwbps18.fa
```
Get the protein sequences and the CDS file too: 

https://pcingola.github.io/SnpEff/snpeff/build_db/#step-3-checking-the-database
```
cd snpEff/data/Hcwbps18
wget https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS18/species/haemonchus_contortus/PRJEB506/haemonchus_contortus.PRJEB506.WBPS18.protein.fa.gz
mv haemonchus_contortus.PRJEB506.WBPS18.protein.fa.gz proteins.fa.gz

wget https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS18/species/haemonchus_contortus/PRJEB506/haemonchus_contortus.PRJEB506.WBPS18.CDS_transcripts.fa.gz
mv haemonchus_contortus.PRJEB506.WBPS18.CDS_transcripts.fa.gz cds.fa.gz
```
Note, if you then try to build the database it complains that it can't find the transcripts/proteins in the fasta files because it's looking for 'transcript:HCON...'. 
So: 
```
zcat cds.fa.gz | sed 's/>/>transcript:/g' > tmp
less tmp
mv tmp cds.fa
gzip cds.fa

zcat protein.fa.gz | sed 's/>/>transcript:/g' > tmp
less tmp
mv tmp protein.fa
gzip protein.fa
```

4. Build database
```
cd snpEff
java -jar snpEff.jar build -v Hcwbps18 
```
Completes quickly in the cmd line! 

To check it is there...
```
java -jar snpEff.jar databases | grep -i contortus
```

# Make VCF file of mutations (snps/indels) of interest: 

Next, you need to have a .vcf file of all the SNPs of interest. To create one can use bcftools:

First, make a new bam.list, with experiment 3 (F3) in line order: 
```
/home/jenni/data_folder/selection_lines_poolseq_bams/3-14949_postCTL3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/1-14031_CTL3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/2-14036_postCTL3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/6-14917_postIVM3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/5-14916_postIVM3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/4-14907_postIVM3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/9-14954_postMOX3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/7-14029_postMOX3.merged.sorted.marked.bam
/home/jenni/data_folder/selection_lines_poolseq_bams/8-14953_postMOX3.merged.sorted.marked.bam
```
```
module load bcftools/1.17

bcftools mpileup -B -q 20 -Q 30 --ff DUP -d 100 -f /home/jenni/data_folder/working_folder/selection_lines_poolseq/mpileup/ref.fa -b /home/jenni/data_folder/working_folder/selection_lines_poolseq/mpileup/expt3/bam.list | bcftools call -mv -O v > HcSL.poolseq.variants.vcf 2>error_vcf.log &

# Note for Tcirc 2017 analysis I used samtools mpileup (v1.3) would have had the depth per sample in it (-D), which my one does not - this is no longer an option in the modern samtools mpileup. It does not appear to be a requirement of the vcf files used by snpeff. I also used to have the option -u (uncompressed vcf,  preferred for piping) - this option is also no longer available in v1.17.
# Now, samtoolsv1.17 mpileup can no longer make vcfs for the bcftools call option - so you have to use bcftools mpileup instead.
# No idea what I mean by this: "Have therefore decided to use my mpileup file that I have already made. Will re scp it across from the HPCC." I have made the vcf file using the bcftools. Probably a comment I forgot to delete. 
# -O v outputs a file in the vcf format
# -mv outputs variant sites only, allowing multiallelic/rare variant calling.
# Note, for this analysis (Hcon SelLps) I have added requirement for -d : the maximum number of integer reads to be read is 100. This is based on the fact that I know that these experiment 3 samples have good coverage depth, and that many sites have coverage of 100. Also, I would expect to identify most variants using this depth (and n = 400 chromosomes). This should speed up the time taken to produce the output file, as higher coverage sites will be subsampled. NOTE however that I do not think this will exclude lower coverage sites from being included. As only the variants are required this will be fine. YES - this is correct - it will subsample higher sites: "Maximum Depth: maximum raw per-file depth. By default, this value is set to 250, as this depth is enough to achieve good results. In this case, if the coverage depth is higher than this threshold, the algorithm will pick up 250 alignments at random. Nevertheless, if you want to tune this value higher, keep in mind that the time of execution will also be higher." from https://manual.omicsbox.biobam.com/user-manual/omicsbox-modules/module-genetic-variation/variant-calling/variant-calling-using-bcftools/#:~:text=BCFtools%20uses%20two%20algorithms.,position%20(%E2%80%9Cpileup%E2%80%9D).
# -B disables the BAQ. Nice discussion on it here: https://www.biostars.org/p/9466154/ 

disown -h jobid
```
Seems to have worked fine. Some have DP (sum of raw read depth of all samples) as > 900 (i.e. 9x100) but this may be counting and then maybe it selects just 100 to use for the SNPs?? Not sure. 

Ok, so I have 18 million+ sites included. 


To run snpeff with ALL variants - can sift them afterwards (with snpsift etc) and I think it will give a fuller picture. Also, if there is a drop out of alleles over the region of interest could have lower coverage there. 


```
java -Xmx4g -jar /home/jenni/data_folder/software/snpEff/snpEff.jar -c /home/jenni/data_folder/software/snpEff/snpEff.config -v -stats Hcwbps18_HcSL_expt3.poolseq.ALL_variants.html Hcwbps18 /home/jenni/data_folder/working_folder/selection_lines_poolseq/mpileup/HcSL.poolseq.variants.vcf > HcSL.poolseq.ALL.variants.ann.vcf 2>error_snpeff_allvariants.log &

```

# RNA-Seq and SnpEff analysis - F3 generation
Using the HcSL.poolseq.ALL.variants.ann.vcf file made above (F3 gen only, but without filtering the vcf before running snpeff), look to identify whether any SNPs are common to affecting DEGs in the F3 adults which Roz used for her RNA-Seq data. 

Note - I had thought to do this (and did so, see this section below), however the aim is undermined by the input gff file. SnpEff will only look for upstream or downstream variants within 5 kb of a gene/annotation, so no single SNP will be recognised to affect many genes in this way.

I subsequently checked what was annotated in the gff file on WBPS - were there miRNAs or spliced leader sequences annotated for example? In other words, were there annotations which I could recognise might affect multiple genes? In short, not that I can see.

```
awk '{print $3}' /home/jenni/data_folder/wbps18/haemonchus_contortus.PRJEB506.WBPS18.annotations.gtf | sort | uniq -c
 191582 CDS
 196915 exon
  16125 five_prime_UTR
  19778 gene
  21319 mRNA
    131 pseudogene
 316607 repeat_region
      2 RNA
      2 rRNA
  56703 tandem_repeat
  17697 three_prime_UTR
     22 tRNA
```

I then looked to see whether any SNPs had an annotation related to the 3' UTR. I used the subsampled SnpEff files created below, which had all lines containing genes of interest. 
```
# For example:
grep -e 'UTR_3' DEGs_Mox_SnpEff.txt > three_prime_UTR_DEGs_Mox_SnpEff.txt
grep -e 'UTR_5' DEGs_Mox_SnpEff.txt > five_prime_UTR_DEGs_Mox_SnpEff.txt
```
No SNPs were identified which were within the annotated 5 or 3 prime UTRs for these genes. Note that the UTR regions may be incomplete, although will likely have been manually curated by Steve Doyle on Apollo. 

```
# Server = scamper
# pwd = /home/jenni/data_folder/SLPS_F3SnpEff_RNASeq_analysis

ln -s /home/jenni/data_folder/software/snpEff/HcSL.poolseq.ALL.variants.ann.vcf HcSL.poolseq.ALL.variants.ann.vcf

# Obtain padj<0.01 ("sig") DEGs identified in both MOX and IVM analyses, and both males and females - 15 genes in total
nano DEGs_IvmMox.txt

#####
HCON_00152760
HCON_00154630
HCON_00154660
HCON_00155240
HCON_00155380
HCON_00155465
HCON_00155650
HCON_00155770
HCON_00155935
HCON_00156060
HCON_00156240
HCON_00156285
HCON_00156400
HCON_00158160
HCON_00158210
#####

# Filter the SnpEff file by these genes, concatenating into one large file:

while IFS= read -r line; do \
grep -e $line ./HcSL.poolseq.ALL.variants.ann.vcf >> DEGs_IvmMox_SnpEff.txt
done < ./DEGs_IvmMox.txt

cat DEGs_IvmMox_SnpEff.txt | awk '{print $2}' | sort -n | uniq -c | sort -k 1 > out # Look for any SNPs which are common to all/most genes
# Only have snps occuring once or, for a few, occuring twice. Therefore there are essentially no snps common to all or even most of these 15 DEGs.
# This therefore also indicates that even looking at just those in IVM or MOX analyses will not identify common snps, or would have had more genes sharing snps.

# However, worth checking whether DEGs specific to Ivm or Mox share common snps.

nano DEGs_Ivm.txt
###
HCON_00151920
###
# As only one, will check and see what SNPs are predicted to affect it. But not sure how to interpret unless very obvious.

grep -e 'HCON_00151920' ./HcSL.poolseq.ALL.variants.ann.vcf > DEGs_Ivm_SnpEff.txt

# Note - many SNPs with potential to affect HCON_00151920. However, the gene itself has a warning: 'TRANSCRIPT INCOMPLETE'. Perhaps worth re-checking it's annotation with RNA-Seq data and ensuring we have the correct protein.

nano DEGs_Mox.txt

# Lots of genes for Mox only (n=48)

while IFS= read -r line; do \
grep -e $line ./HcSL.poolseq.ALL.variants.ann.vcf >> DEGs_Mox_SnpEff.txt
done < ./DEGs_Mox.txt

cat DEGs_Mox_SnpEff.txt | awk '{print $2}' | sort -n | uniq -c | sort -k 1 > out # Look for any SNPs which are common to all/most genes
# There are 4 SNPs which occur 4 times, the rest occur less commonly than this. Out of a total of 48 genes that is not very many.
      4 37150064
      4 42317729
      4 42317826
      4 42318889

grep -e '37150064' DEGs_Mox_SnpEff.txt > SNP_37150064_DEGs_Mox_SnpEff.txt

grep -e '42317729' DEGs_Mox_SnpEff.txt > SNP_42317729_DEGs_Mox_SnpEff.txt

grep -e '42317826' DEGs_Mox_SnpEff.txt > SNP_42317826_DEGs_Mox_SnpEff.txt

grep -e '42318889' DEGs_Mox_SnpEff.txt > SNP_42318889_DEGs_Mox_SnpEff.txt

# The 1st is (but it has actually been pulled out just twice - but it has both a SNP and an indel at the same position):

hcontortus_chr5_Celeg_TT_arrow_pilon    37150064        .       A       G       967.092 .       DP=705;VDB=2.49495e-06;SGB=-244.128;RPBZ=-2.67695;MQBZ=-5.60072;MQSBZ=-2.29108;BQBZ=-0.721372;SCBZ=1.40767;MQ0F=0;AC=4;AN=18;DP4=318,145,54,61;MQ=59;
ANN=
G|upstream_gene_variant|MODIFIER|HCON_00155070|gene:HCON_00155070|transcript|transcript:HCON_00155070-00001|protein_coding||c.-29T>C|||||29|,
G|upstream_gene_variant|MODIFIER|HCON_00155080|gene:HCON_00155080|transcript|transcript:HCON_00155080-00001|protein_coding||c.-4475A>G|||||4272|,
G|upstream_gene_variant|MODIFIER|HCON_00155080|gene:HCON_00155080|transcript|transcript:HCON_00155080-00002|protein_coding||c.-4475A>G|||||4272|,
G|downstream_gene_variant|MODIFIER|HCON_00155060|gene:HCON_00155060|transcript|transcript:HCON_00155060-00001|protein_coding||c.*4888A>G|||||4845|,
G|intergenic_region|MODIFIER|HCON_00155070-HCON_00155080|gene:HCON_00155070-gene:HCON_00155080|intergenic_region|gene:HCON_00155070-gene:HCON_00155080|||n.37150064A>G||||||
GT:PL   0/1:255,0,255   0/1:255,0,255       0/1:255,0,255   0/0:0,5,255     0/0:0,21,255    0/0:0,45,255    0/1:255,0,255   0/0:0,116,255   0/0:0,88,255

hcontortus_chr5_Celeg_TT_arrow_pilon    37150064        .       AAG     AAGAG   505.503 .       INDEL;IDV=30;IMF=0.441176;DP=705;VDB=0.0224376;SGB=-234.274;RPBZ=-5.2879;MQBZ=1.41114;MQSBZ=-2.29108;BQBZ=-0.721372;SCBZ=-8.86764;MQ0F=0;AC=3;AN=18;DP4=313,183,31,14;MQ=59;
ANN=
AAGAG|upstream_gene_variant|MODIFIER|HCON_00155070|gene:HCON_00155070|transcript|transcript:HCON_00155070-00001|protein_coding||c.-32_-31insCT|||||32|,
AAGAG|upstream_gene_variant|MODIFIER|HCON_00155080|gene:HCON_00155080|transcript|transcript:HCON_00155080-00001|protein_coding||c.-4473_-4472insAG|||||4269|,
AAGAG|upstream_gene_variant|MODIFIER|HCON_00155080|gene:HCON_00155080|transcript|transcript:HCON_00155080-00002|protein_coding||c.-4473_-4472insAG|||||4269|,
AAGAG|downstream_gene_variant|MODIFIER|HCON_00155060|gene:HCON_00155060|transcript|transcript:HCON_00155060-00001|protein_coding||c.*4890_*4891insAG|||||4848|,
AAGAG|intergenic_region|MODIFIER|HCON_00155070-HCON_00155080|gene:HCON_00155070-gene:HCON_00155080|intergenic_region|gene:HCON_00155070-gene:HCON_00155080|||n.37150066_37150067insAG||||||     GT:PL   0/1:172,0,41    0/1:155,0,54    0/1:228,0,32    0/0:0,130,46    0/0:0,190,89    0/0:0,148,68    0/0:8,101,0         0/0:0,149,80    0/0:0,196,104

# Note that it is the control samples that are all predicted to be heterozygous for the SNP, while all IVM are homREF and the mox are predicted to be Het/HomREF/HomREF. For the indel, only the control sampls are predicted to be heterozygous, and the rest HomREF. So an odd SNP/indel to be important in drug resistance.

# The genes involved in the MOX DEG list are HCON_00155080 (a STAS domain-containing protein, involved in ion channel transport) and HCON_00155060.

# The 2nd has also only actually been pulled out twice - it has both a SNP and indel at the same position. Again, controls are het and selected are homREF.

# The 3rd again has both a SNP and an indel. Again het/homREF is somewhat random.

# The 4th is the same as the others.
```
So, conclusion is that there is no single mutation, or even vaguely common mutation, driving this gene dysregulation that is apparent in the reference assembly. 

To check for genes in the lists which have warnings regarding their annotations. In case Roz wants them. 
```
for i in D*SnpEff.txt ;
do
awk 'BEGIN {FS=";"} {OFS="\t"} {print $14}' ${i} | tr ',' '\n' | grep -e 'WARNING' |awk 'BEGIN {FS="|"} {OFS="\t"} {print $4, $16}' | awk 'BEGIN {FS="GT:PL"} {OFS="\t"} {print $1}' | sed 's/\t$//g' | sort | uniq > ${i}_WARNINGS ;
done &

# Then, to identify which of these are in the DEG list:

cat DEGs*WARNINGS > All_WARNINGS_genes
cat DEGs_IvmMox.txt DEGs_Ivm.txt DEGs_Mox.txt > All_DEGs.txt

while IFS= read -r line; do \
grep -e $line ./All_WARNINGS_genes >> DEGs_with_WARNINGS.txt 
done < ./All_DEGs.txt

cat DEGs_with_WARNINGS.txt | sort | uniq > tmp
mv tmp DEGs_with_WARNINGS.txt
```

```
# DEGs with warnings in annotation
HCON_00151920   WARNING_TRANSCRIPT_INCOMPLETE
HCON_00155465   WARNING_TRANSCRIPT_NO_START_CODON
HCON_00155935   WARNING_TRANSCRIPT_NO_START_CODON
HCON_00154780   WARNING_TRANSCRIPT_NO_START_CODON
HCON_00157770   WARNING_TRANSCRIPT_NO_START_CODON
HCON_00159960   WARNING_TRANSCRIPT_NO_START_CODON
HCON_00160220   WARNING_TRANSCRIPT_NO_START_CODON
HCON_00161470   WARNING_TRANSCRIPT_NO_START_CODON
```
Done.
