**For each sample, 200 L3 were pooled to extract gDNA.**

LIMS ID: LIMS28593 and LIMS29763

Platform: NovaSeq

```
wget --no-check-certificate -r --cut-dirs=2 -np -nH -R "index.html*" link/to/files # This will download all the trimmed read files in one go. Super!

# Certificate had expired and the error message suggested inserting --no-check-certificate to be able to download.

##########################################################
## FINISHED --2022-11-30 17:33:11--                     ##
## Downloaded: 117 files, 733G in 7h 22m 0s (28.3 MB/s) ##
##########################################################
```

"For paired-end sequence data, there are three sequence file types. The files labelled R1 and R2 contain the corresponding paired-end sequences. The singlet files contain sequences whose pair has been removed due to poor sequence quality or adapter contamination. If a sample has been sequenced several times, there will be several sets of sequence files in the sample directory. These will need to be concatenated before downstream analysis."



Summary Statistics
Diagram illustrating the total number of reads obtained for each sample.
![image](https://user-images.githubusercontent.com/55552826/204764051-e6828d97-6a82-4fb4-8ca7-5b1db1ed2069.png)


Box plot showing the distribution of trimmed read lengths for the forward (R1), reverse (R2) and singlet (R0) reads. Note that it is common for a small number of reads to consist of mostly adapter-derived sequence, so it is expected that the distribution will show a long tail.
![image](https://user-images.githubusercontent.com/55552826/204764091-3d646558-dcea-4edd-b740-49e4363d9ed1.png)
![image](https://user-images.githubusercontent.com/55552826/204764116-d75db488-e51f-4b75-b1e3-fc23bf08bfcf.png)


Further detailed statistics are available for each of the trimmed and raw data files.


# Description of methods
The raw Fastq files are trimmed for the presence of Illumina adapter sequences using Cutadapt version 1.2.1 [Reference]. The option -O 3 was used, so the 3' end of any reads which match the adapter sequence for 3 bp. or more are trimmed.

The reads are further trimmed using Sickle version 1.200 with a minimum window quality score of 20. Reads shorter than 15 bp. after trimming were removed. If only one of a read pair passed this filter, it is included in the R0 file. The output files from Cutadapt and Sickle are available here.

Statistics were generated using fastq-stats from EAUtils.

Additional notes
Libraries by Rosie Crompton ; Sequencing by Edith Vamos.

Contacts
The bioinformatics analysis for this project has been performed by R.Gregory. Please e-mail if you have any further queries.


# RNA-Seq samples

Transferred the trimmed reads from the liverpool website to the Glasgow HPCC ~/lainglab/HCON/Selection_lines_RNAseq_IVMMOX_2023/trimmed_data/Trimmed  on 12th Feb 2024

```
FINISHED --2024-02-12 15:30:31--
Total wall clock time: 2h 33m 2s
Downloaded: 73 files, 106G in 2h 31m 10s (12.0 MB/s)
```
Correct number of files: 18 * 3 for the fastq (54) plus 18 sample directories plus the 'Trimmed' directory. 
